--- BlockSnow.java
+++ BlockSnow.java
@@ -9,15 +9,15 @@
         this.setTickOnLoad(true);
     }
-    
+
     @Override
     public AxisAlignedBB getCollisionBoundingBoxFromPool(final World world, final int x, final int y, final int z) {
         return null;
     }
-    
+
     @Override
     public boolean isOpaqueCube() {
         return false;
     }
-    
+
     @Override
     public boolean canPlaceBlockAt(final World world, final int x, final int y, final int z) {
@@ -25,10 +25,10 @@
         return i5 != 0 && Block.blocksList[i5].isOpaqueCube() && world.getBlockMaterial(x, y - 1, z).getIsSolid();
     }
-    
+
     @Override
     public void onNeighborBlockChange(final World world, final int x, final int y, final int z, final int flag) {
         this.canSnowStay(world, x, y, z);
     }
-    
+
     private boolean canSnowStay(final World world, final int x, final int y, final int z) {
         if (!this.canPlaceBlockAt(world, x, y, z)) {
@@ -39,15 +39,15 @@
         return true;
     }
-    
+
     @Override
     public int idDropped(final int count, final Random random) {
         return Item.snowball.shiftedIndex;
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
         return 0;
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
@@ -57,5 +57,5 @@
         }
     }
-    
+
     @Override
     public boolean shouldSideBeRendered(final IBlockAccess blockAccess, final int x, final int y, final int z, final int side) {
