--- BlockSand.java
+++ BlockSand.java
@@ -1,4 +1,6 @@
 package net.minecraft.src;
 
+import org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason;
+
 import java.util.Random;
 
@@ -38,5 +40,5 @@
             }
             else {
-                world.spawnEntityInWorld(entityFallingSand8);
+                world.spawnEntityInWorld(entityFallingSand8, SpawnReason.NATURAL);
             }
         }
@@ -53,5 +55,5 @@
             return true;
         }
-        if (i4 == Block.fire.blockID) {
+        if (i4 == Block.FIRE.blockID) {
             return true;
         }
