--- BlockLadder.java
+++ BlockLadder.java
@@ -7,5 +7,5 @@
         super(id, blockIndex, Material.circuits);
     }
-    
+
     @Override
     public AxisAlignedBB getCollisionBoundingBoxFromPool(final World world, final int x, final int y, final int z) {
@@ -26,20 +26,20 @@
         return super.getCollisionBoundingBoxFromPool(world, x, y, z);
     }
-    
+
     @Override
     public boolean isOpaqueCube() {
         return false;
     }
-    
+
     @Override
     public int getRenderType() {
         return 8;
     }
-    
+
     @Override
     public boolean canPlaceBlockAt(final World world, final int x, final int y, final int z) {
         return world.isBlockNormalCube(x - 1, y, z) || world.isBlockNormalCube(x + 1, y, z) || world.isBlockNormalCube(x, y, z - 1) || world.isBlockNormalCube(x, y, z + 1);
     }
-    
+
     @Override
     public void onBlockPlaced(final World world, final int x, final int y, final int z, final int notifyFlag) {
@@ -59,5 +59,5 @@
         world.setBlockMetadataWithNotify(x, y, z, i6);
     }
-    
+
     @Override
     public void onNeighborBlockChange(final World world, final int x, final int y, final int z, final int flag) {
@@ -82,5 +82,5 @@
         super.onNeighborBlockChange(world, x, y, z, flag);
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
