--- BlockSafe.java
+++ BlockSafe.java
@@ -1,12 +1,9 @@
 package net.minecraft.src;
 
-import net.minecraft.src.EntityPlayer;
-import net.minecraft.src.World;
-
 public class BlockSafe extends BlockMultiSided {
     protected BlockSafe(final int n, final int n2, final int n3, final int n4) {
         super(n, n2, n3, n4);
     }
-    
+
     @Override
     public boolean blockActivated(final World world, final int x, final int y, final int z, final EntityPlayer entityPlayer) {
