--- BlockSand.java
+++ BlockSand.java
@@ -1,4 +1,6 @@
 package net.minecraft.src;
 
+import org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason;
+
 import java.util.Random;
 
@@ -38,5 +40,5 @@
             }
             else {
-                world.spawnEntityInWorld(entityFallingSand8);
+                world.spawnEntityInWorld(entityFallingSand8, SpawnReason.NATURAL);
             }
         }
