--- BlockTNT.java
+++ BlockTNT.java
@@ -7,10 +7,10 @@
         super(id, blockIndex, Material.tnt);
     }
-    
+
     @Override
     public int getBlockTextureFromSide(final int side) {
         return (side == 0) ? (this.blockIndexInTexture + 2) : ((side == 1) ? (this.blockIndexInTexture + 1) : this.blockIndexInTexture);
     }
-    
+
     @Override
     public void onNeighborBlockChange(final World world, final int x, final int y, final int z, final int flag) {
@@ -20,10 +20,10 @@
         }
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
         return 0;
     }
-    
+
     @Override
     public void onBlockDestroyedByExplosion(final World world, final int x, final int y, final int z) {
@@ -32,10 +32,10 @@
         world.spawnEntityInWorld(entityTNTPrimed5);
     }
-    
+
     @Override
     public void onBlockDestroyedByPlayer(final World world, final int x, final int y, final int z, final int flag) {
         final EntityTNTPrimed entityTNTPrimed6 = new EntityTNTPrimed(world, x + 0.5f, y + 0.5f, z + 0.5f);
         world.spawnEntityInWorld(entityTNTPrimed6);
-        world.playSoundAtEntity((Entity)entityTNTPrimed6, "random.fuse", 1.0f, 1.0f);
+        world.playSoundAtEntity((Entity) entityTNTPrimed6, "random.fuse", 1.0f, 1.0f);
     }
 }
