--- BlockLog.java
+++ BlockLog.java
@@ -8,15 +8,15 @@
         this.blockIndexInTexture = 20;
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
         return 1;
     }
-    
+
     @Override
     public int idDropped(final int count, final Random random) {
         return Block.wood.blockID;
     }
-    
+
     @Override
     public int getBlockTextureFromSide(final int side) {
