--- BlockGrass.java
+++ BlockGrass.java
@@ -9,5 +9,5 @@
         this.setTickOnLoad(true);
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
@@ -17,6 +17,5 @@
             }
             world.setBlockWithNotify(x, y, z, Block.dirt.blockID);
-        }
-        else if (world.getBlockLightValue(x, y + 1, z) >= 9) {
+        } else if (world.getBlockLightValue(x, y + 1, z) >= 9) {
             final int i6 = x + random.nextInt(3) - 1;
             final int i7 = y + random.nextInt(5) - 3;
@@ -27,5 +26,5 @@
         }
     }
-    
+
     @Override
     public int idDropped(final int count, final Random random) {
