--- BlockFire.java
+++ BlockFire.java
@@ -1,4 +1,11 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.event.block.BlockBurnEvent;
+import org.bukkit.event.block.BlockIgniteEvent;
+import org.bukkit.event.block.BlockIgniteEvent.IgniteCause;
+import org.bukkit.event.block.BlockSpreadEvent;
+import org.bukkit.material.MaterialData;
+
 import java.util.Random;
 
@@ -72,4 +79,13 @@
             this.tryToCatchBlockOnFire(world, x, y, z - 1, 300, random);
             this.tryToCatchBlockOnFire(world, x, y, z + 1, 300, random);
+
+            // CraftBukkit start - Call to stop spread of fire.
+            org.bukkit.Server server = Bukkit.getServer();
+            org.bukkit.World bworld = server.getWorld(world.levelName);
+
+            IgniteCause igniteCause = BlockIgniteEvent.IgniteCause.SPREAD;
+            org.bukkit.block.Block fromBlock = bworld.getBlockAt(x, y, z);
+            // CraftBukkit end
+
             for (int i7 = x - 1; i7 <= x + 1; ++i7) {
                 for (int i8 = z - 1; i8 <= z + 1; ++i8) {
@@ -82,5 +98,28 @@
                             final int i11 = this.getChanceOfNeighborsEncouragingFire(world, i7, i9, i8);
                             if (i11 > 0 && random.nextInt(i10) <= i11) {
-                                world.setBlockWithNotify(i7, i9, i8, this.blockID);
+                                //world.setBlockWithNotify(i7, i9, i8, this.blockID);
+                                // CraftBukkit start - Call to stop spread of fire.
+                                org.bukkit.block.Block block = bworld.getBlockAt(i7, i9, i8);
+
+                                if (block.getTypeId() != Block.fire.blockID) {
+                                    BlockIgniteEvent event = new BlockIgniteEvent(block, igniteCause, null);
+                                    server.getPluginManager().callEvent(event);
+
+                                    if (event.isCancelled()) {
+                                        continue;
+                                    }
+
+                                    org.bukkit.block.BlockState blockState = bworld.getBlockAt(i7, i9, i8).getState();
+                                    blockState.setTypeId(this.blockID);
+                                    //blockState.setData(new MaterialData(this.blockID, (byte) k2));
+
+                                    BlockSpreadEvent spreadEvent = new BlockSpreadEvent(blockState.getBlock(), fromBlock, blockState);
+                                    server.getPluginManager().callEvent(spreadEvent);
+
+                                    if (!spreadEvent.isCancelled()) {
+                                        blockState.update(true);
+                                    }
+                                }
+                                // CraftBukkit end
                             }
                         }
@@ -95,4 +134,14 @@
         if (random.nextInt(chance) < i7) {
             final boolean z2 = world.getBlockId(x, y, z) == Block.tnt.blockID;
+            // CraftBukkit start
+            org.bukkit.block.Block theBlock = Bukkit.getWorld(world.levelName).getBlockAt(x, y, z);
+
+            BlockBurnEvent event = new BlockBurnEvent(theBlock);
+            Bukkit.getServer().getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                return;
+            }
+            // CraftBukkit end
             if (random.nextInt(2) == 0) {
                 world.setBlockWithNotify(x, y, z, this.blockID);
