--- BlockCactus.java
+++ BlockCactus.java
@@ -1,4 +1,8 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.event.entity.EntityDamageByBlockEvent;
+import org.bukkit.event.entity.EntityDamageEvent;
+
 import java.util.Random;
 
@@ -76,9 +80,24 @@
         }
         final int i5 = world.getBlockId(x, y - 1, z);
-        return i5 == Block.cactus.blockID || i5 == Block.sand.blockID;
+        return i5 == Block.CACTUS.blockID || i5 == Block.SAND.blockID;
     }
     
     @Override
     public void onEntityCollidedWithBlock(final World world, final int x, final int y, final int z, final Entity entity) {
+        // CraftBukkit start - ENTITY_DAMAGEBY_BLOCK event
+        if (entity instanceof EntityLiving) {
+            org.bukkit.block.Block damager = Bukkit.getWorld(world.levelName).getBlockAt(x, y, z);
+            org.bukkit.entity.Entity damagee = entity.getBukkitEntity();
+
+            EntityDamageByBlockEvent event = new EntityDamageByBlockEvent(damager, damagee, EntityDamageEvent.DamageCause.CONTACT, 1);
+            Bukkit.getServer().getPluginManager().callEvent(event);
+
+            if (!event.isCancelled()) {
+                entity.attackEntityFrom(null, event.getDamage());
+            }
+            return;
+        }
+        // CraftBukkit end
+
         entity.attackEntityFrom(null, 1);
     }
