--- BlockFlowing.java
+++ BlockFlowing.java
@@ -1,4 +1,8 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.block.BlockFace;
+import org.bukkit.event.block.BlockFromToEvent;
+
 import java.util.Random;
 
@@ -7,5 +11,5 @@
     boolean[] isOptimalFlowDirection;
     int[] flowCost;
-    
+
     protected BlockFlowing(final int id, final Material material) {
         super(id, material);
@@ -14,5 +18,5 @@
         this.flowCost = new int[4];
     }
-    
+
     private void updateFlow(final World world, final int x, final int y, final int z) {
         final int i5 = world.getBlockMetadata(x, y, z);
@@ -21,7 +25,13 @@
         world.markBlockNeedsUpdate(x, y, z);
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
+        // CraftBukkit start
+        org.bukkit.World bworld = Bukkit.getWorld(world.levelName);
+        org.bukkit.Server server = Bukkit.getServer();
+        org.bukkit.block.Block source = bworld == null ? null : bworld.getBlockAt(x, y, z);
+        // CraftBukkit end
+
         int i6 = this.getFlowDecay(world, x, y, z);
         boolean z2 = true;
@@ -41,6 +51,5 @@
                 if (i9 >= 8) {
                     i8 = i9;
-                }
-                else {
+                } else {
                     i8 = i9 + 8;
                 }
@@ -49,6 +58,5 @@
                 if (world.isBlockNormalCube(x, y - 1, z)) {
                     i8 = 0;
-                }
-                else if (world.getBlockMaterial(x, y - 1, z) == this.material && world.getBlockMetadata(x, y, z) == 0) {
+                } else if (world.getBlockMaterial(x, y - 1, z) == this.material && world.getBlockMetadata(x, y, z) == 0) {
                     i8 = 0;
                 }
@@ -61,27 +69,37 @@
                 if ((i6 = i8) < 0) {
                     world.setBlockWithNotify(x, y, z, 0);
-                }
-                else {
+                } else {
                     world.setBlockMetadataWithNotify(x, y, z, i8);
                     world.scheduleBlockUpdate(x, y, z, this.blockID);
                     world.notifyBlocksOfNeighborChange(x, y, z, this.blockID);
                 }
-            }
-            else if (z2) {
+            } else if (z2) {
                 this.updateFlow(world, x, y, z);
             }
-        }
-        else {
+        } else {
             this.updateFlow(world, x, y, z);
         }
         if (this.liquidCanDisplaceBlock(world, x, y - 1, z)) {
-            if (i6 >= 8) {
+            /*if (i6 >= 8) {
                 world.setBlockAndMetadataWithNotify(x, y - 1, z, this.blockID, i6);
             }
             else {
                 world.setBlockAndMetadataWithNotify(x, y - 1, z, this.blockID, i6 + 8);
+            }*/
+            // CraftBukkit start - send "down" to the server
+            BlockFromToEvent event = new BlockFromToEvent(source, BlockFace.DOWN);
+            if (server != null) {
+                server.getPluginManager().callEvent(event);
             }
-        }
-        else if (i6 >= 0 && (i6 == 0 || this.blockBlocksFlow(world, x, y - 1, z))) {
+
+            if (!event.isCancelled()) {
+                if (i6 >= 8) {
+                    world.setBlockAndMetadataWithNotify(x, y - 1, z, this.blockID, i6);
+                } else {
+                    world.setBlockAndMetadataWithNotify(x, y - 1, z, this.blockID, i6 + 8);
+                }
+            }
+            // CraftBukkit end
+        } else if (i6 >= 0 && (i6 == 0 || this.blockBlocksFlow(world, x, y - 1, z))) {
             final boolean[] z3 = this.getOptimalFlowDirections(world, x, y, z);
             int i8 = i6 + this.fluidType;
@@ -92,5 +110,5 @@
                 return;
             }
-            if (z3[0]) {
+            /*if (z3[0]) {
                 this.flowIntoBlock(world, x - 1, y, z, i8);
             }
@@ -103,8 +121,28 @@
             if (z3[3]) {
                 this.flowIntoBlock(world, x, y, z + 1, i8);
+            }*/
+
+            // CraftBukkit start - all four cardinal directions. Do not change the order!
+            BlockFace[] faces = new BlockFace[] { BlockFace.NORTH, BlockFace.SOUTH, BlockFace.EAST, BlockFace.WEST };
+            int index = 0;
+
+            for (BlockFace currentFace: faces) {
+                if (z3[index]) {
+                    BlockFromToEvent event = new BlockFromToEvent(source, currentFace);
+
+                    if (server != null) {
+                        server.getPluginManager().callEvent(event);
+                    }
+
+                    if (!event.isCancelled()) {
+                        this.flowIntoBlock(world, x + currentFace.getModX(), y, z + currentFace.getModZ(), i8);
+                    }
+                }
+                index++;
             }
+            // CraftBukkit end
         }
     }
-    
+
     private void flowIntoBlock(final World world, final int x, final int y, final int z, final int metadata) {
         if (this.liquidCanDisplaceBlock(world, x, y, z)) {
@@ -113,6 +151,5 @@
                 if (this.material == Material.lava) {
                     this.triggerLavaMixEffects(world, x, y, z);
-                }
-                else {
+                } else {
                     Block.blocksList[i6].dropBlockAsItem(world, x, y, z, world.getBlockMetadata(x, y, z));
                 }
@@ -121,5 +158,5 @@
         }
     }
-    
+
     private int calculateFlowCost(final World world, final int x, final int y, final int z, final int i5, final int i6) {
         int i7 = 1000;
@@ -155,5 +192,5 @@
         return i7;
     }
-    
+
     private boolean[] getOptimalFlowDirections(final World world, final int x, final int y, final int z) {
         for (int i5 = 0; i5 < 4; ++i5) {
@@ -176,6 +213,5 @@
                 if (!this.blockBlocksFlow(world, i6, y - 1, i7)) {
                     this.flowCost[i5] = 0;
-                }
-                else {
+                } else {
                     this.flowCost[i5] = this.calculateFlowCost(world, i6, y, i7, 1, i5);
                 }
@@ -193,5 +229,5 @@
         return this.isOptimalFlowDirection;
     }
-    
+
     private boolean blockBlocksFlow(final World world, final int x, final int y, final int z) {
         final int i5 = world.getBlockId(x, y, z);
@@ -205,5 +241,5 @@
         return material6.isSolid();
     }
-    
+
     protected int getSmallestFlowDecay(final World world, final int x, final int y, final int z, final int i5) {
         int i6 = this.getFlowDecay(world, x, y, z);
@@ -219,10 +255,10 @@
         return (i5 >= 0 && i6 >= i5) ? i5 : i6;
     }
-    
+
     private boolean liquidCanDisplaceBlock(final World world, final int x, final int y, final int z) {
         final Material material5 = world.getBlockMaterial(x, y, z);
         return material5 != this.material && material5 != Material.lava && !this.blockBlocksFlow(world, x, y, z);
     }
-    
+
     @Override
     public void onBlockAdded(final World world, final int x, final int y, final int z) {
