--- BlockRedstoneOre.java
+++ BlockRedstoneOre.java
@@ -5,5 +5,5 @@
 public class BlockRedstoneOre extends Block {
     private boolean glowing;
-    
+
     public BlockRedstoneOre(final int id, final int blockIndex, final boolean glowing) {
         super(id, blockIndex, Material.rock);
@@ -13,10 +13,10 @@
         this.glowing = glowing;
     }
-    
+
     @Override
     public int tickRate() {
         return 30;
     }
-    
+
     @Override
     public void onBlockClicked(final World world, final int x, final int y, final int z, final EntityPlayer entityPlayer) {
@@ -24,5 +24,5 @@
         super.onBlockClicked(world, x, y, z, entityPlayer);
     }
-    
+
     @Override
     public void onEntityWalking(final World world, final int x, final int y, final int z, final Entity entity) {
@@ -30,5 +30,5 @@
         super.onEntityWalking(world, x, y, z, entity);
     }
-    
+
     @Override
     public boolean blockActivated(final World world, final int x, final int y, final int z, final EntityPlayer entityPlayer) {
@@ -36,5 +36,5 @@
         return super.blockActivated(world, x, y, z, entityPlayer);
     }
-    
+
     private void glow(final World world, final int x, final int y, final int z) {
         this.sparkle(world, x, y, z);
@@ -43,5 +43,5 @@
         }
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
@@ -50,15 +50,15 @@
         }
     }
-    
+
     @Override
     public int idDropped(final int count, final Random random) {
         return Item.redstone.shiftedIndex;
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
         return 4 + random.nextInt(2);
     }
-    
+
     private void sparkle(final World world, final int x, final int y, final int z) {
         final Random random5 = world.rand;
