--- BlockLeavesBase.java
+++ BlockLeavesBase.java
@@ -3,15 +3,15 @@
 public class BlockLeavesBase extends Block {
     protected boolean graphicsLevel;
-    
+
     protected BlockLeavesBase(final int id, final int blockIndex, final Material material, final boolean graphicsLevel) {
         super(id, blockIndex, material);
         this.graphicsLevel = graphicsLevel;
     }
-    
+
     @Override
     public boolean isOpaqueCube() {
         return false;
     }
-    
+
     @Override
     public boolean shouldSideBeRendered(final IBlockAccess blockAccess, final int x, final int y, final int z, final int side) {
