--- BlockSand.java
+++ BlockSand.java
@@ -5,28 +5,40 @@
 public class BlockSand extends Block {
     public static boolean fallInstantly;
-    
+
     static {
         BlockSand.fallInstantly = false;
     }
-    
+
     public BlockSand(final int id, final int blockIndex) {
         super(id, blockIndex, Material.sand);
     }
-    
+
+    public static boolean canFallBelow(final World world, final int x, final int y, final int z) {
+        final int i4 = world.getBlockId(x, y, z);
+        if (i4 == 0) {
+            return true;
+        }
+        if (i4 == Block.fire.blockID) {
+            return true;
+        }
+        final Material material5 = Block.blocksList[i4].material;
+        return material5 == Material.water || material5 == Material.lava;
+    }
+
     @Override
     public void onBlockAdded(final World world, final int x, final int y, final int z) {
         world.scheduleBlockUpdate(x, y, z, this.blockID);
     }
-    
+
     @Override
     public void onNeighborBlockChange(final World world, final int x, final int y, final int z, final int flag) {
         world.scheduleBlockUpdate(x, y, z, this.blockID);
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
         this.tryToFall(world, x, y, z);
     }
-    
+
     private void tryToFall(final World world, final int x, final int y, final int z) {
         if (canFallBelow(world, x, y - 1, z) && y >= 0) {
@@ -36,26 +48,13 @@
                     entityFallingSand8.onUpdate();
                 }
-            }
-            else {
+            } else {
                 world.spawnEntityInWorld(entityFallingSand8);
             }
         }
     }
-    
+
     @Override
     public int tickRate() {
         return 3;
     }
-    
-    public static boolean canFallBelow(final World world, final int x, final int y, final int z) {
-        final int i4 = world.getBlockId(x, y, z);
-        if (i4 == 0) {
-            return true;
-        }
-        if (i4 == Block.fire.blockID) {
-            return true;
-        }
-        final Material material5 = Block.blocksList[i4].material;
-        return material5 == Material.water || material5 == Material.lava;
-    }
 }
