--- BlockFlower.java
+++ BlockFlower.java
@@ -11,14 +11,14 @@
         this.setBlockBounds(0.5f - f3, 0.0f, 0.5f - f3, 0.5f + f3, f3 * 3.0f, 0.5f + f3);
     }
-    
+
     @Override
     public boolean canPlaceBlockAt(final World world, final int x, final int y, final int z) {
         return this.canThisPlantGrowOnThisBlockID(world.getBlockId(x, y - 1, z));
     }
-    
+
     protected boolean canThisPlantGrowOnThisBlockID(final int id) {
         return id == Block.grass.blockID || id == Block.dirt.blockID || id == Block.tilledField.blockID;
     }
-    
+
     @Override
     public void onNeighborBlockChange(final World world, final int x, final int y, final int z, final int flag) {
@@ -26,10 +26,10 @@
         this.g(world, x, y, z);
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
         this.g(world, x, y, z);
     }
-    
+
     protected final void g(final World world1, final int i2, final int i3, final int i4) {
         if (!this.canBlockStay(world1, i2, i3, i4)) {
@@ -38,20 +38,20 @@
         }
     }
-    
+
     @Override
     public boolean canBlockStay(final World world, final int x, final int y, final int z) {
         return (world.getBlockLightValue(x, y, z) >= 8 || world.canBlockSeeTheSky(x, y, z)) && this.canThisPlantGrowOnThisBlockID(world.getBlockId(x, y - 1, z));
     }
-    
+
     @Override
     public AxisAlignedBB getCollisionBoundingBoxFromPool(final World world, final int x, final int y, final int z) {
         return null;
     }
-    
+
     @Override
     public boolean isOpaqueCube() {
         return false;
     }
-    
+
     @Override
     public int getRenderType() {
