--- BlockOre.java
+++ BlockOre.java
@@ -7,10 +7,10 @@
         super(id, blockIndex, Material.rock);
     }
-    
+
     @Override
     public int idDropped(final int count, final Random random) {
         return (this.blockID == Block.oreCoal.blockID) ? Item.coal.shiftedIndex : ((this.blockID == Block.oreDiamond.blockID) ? Item.diamond.shiftedIndex : this.blockID);
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
