--- BlockMultiSided.java
+++ BlockMultiSided.java
@@ -2,8 +2,4 @@
 
 import java.util.Random;
-import net.minecraft.src.World;
-import net.minecraft.src.IBlockAccess;
-import net.minecraft.src.Material;
-import net.minecraft.src.Block;
 
 public class BlockMultiSided extends Block {
@@ -12,5 +8,5 @@
     private int texBottom;
     private int id;
-    
+
     protected BlockMultiSided(final int n, final int n2, final int n3, final int n4) {
         super(n, Material.grass);
@@ -22,5 +18,5 @@
         this.id = n;
     }
-    
+
     @Override
     public int getBlockTexture(final IBlockAccess iBlockAccess, final int n, final int n2, final int n3, final int n4) {
@@ -33,9 +29,9 @@
         return this.texSide;
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
     }
-    
+
     @Override
     public int idDropped(final int count, final Random random) {
