--- BlockIce.java
+++ BlockIce.java
@@ -9,10 +9,10 @@
         this.setTickOnLoad(true);
     }
-    
+
     @Override
     public boolean shouldSideBeRendered(final IBlockAccess blockAccess, final int x, final int y, final int z, final int side) {
         return super.shouldSideBeRendered(blockAccess, x, y, z, 1 - side);
     }
-    
+
     @Override
     public void onBlockRemoval(final World world, final int x, final int y, final int z) {
@@ -22,10 +22,10 @@
         }
     }
-    
+
     @Override
     public int quantityDropped(final Random random) {
         return 0;
     }
-    
+
     @Override
     public void updateTick(final World world, final int x, final int y, final int z, final Random random) {
