--- ChunkCache.java
+++ ChunkCache.java
@@ -6,5 +6,5 @@
     private Chunk[][] chunkArray;
     private World worldObj;
-    
+
     public ChunkCache(final World world, final int x, final int i3, final int z, final int i5, final int i6, final int i7) {
         this.worldObj = world;
@@ -20,5 +20,5 @@
         }
     }
-    
+
     public int getBlockId(final int blockX, final int blockY, final int blockZ) {
         if (blockY < 0) {
@@ -32,5 +32,5 @@
         return this.chunkArray[i4][i5].getBlockID(blockX & 0xF, blockY, blockZ & 0xF);
     }
-    
+
     public int getBlockMetadata(final int x, final int y, final int z) {
         if (y < 0) {
@@ -44,10 +44,10 @@
         return this.chunkArray[i4][i5].getBlockMetadata(x & 0xF, y, z & 0xF);
     }
-    
+
     public Material getBlockMaterial(final int x, final int y, final int z) {
         final int i4 = this.getBlockId(x, y, z);
         return (i4 == 0) ? Material.air : Block.blocksList[i4].material;
     }
-    
+
     public boolean isBlockNormalCube(final int x, final int y, final int z) {
         final Block block4 = Block.blocksList[this.getBlockId(x, y, z)];
