--- EntityPig.java
+++ EntityPig.java
@@ -1,7 +1,13 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Entity;
+import ru.vladthemountain.lilybukkit.core.entity.LBPig;
+
 public class EntityPig extends EntityAnimal {
     public boolean saddled;
-    
+
+    private final LBPig lbPig; //LilyBukkit
+
     public EntityPig(final World world) {
         super(world);
@@ -10,6 +16,8 @@
         this.setSize(0.9f, 0.9f);
         this.saddled = false;
+
+        this.lbPig = new LBPig(Bukkit.getWorld(world.levelName), this); //LilyBukkit
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
@@ -17,5 +25,5 @@
         nbttagcompound.setBoolean("Saddle", this.saddled);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
@@ -23,23 +31,26 @@
         this.saddled = nbttagcompound.getBoolean("Saddle");
     }
-    
+
     @Override
     protected String getLivingSound() {
         return "mob.pig";
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.pig";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.pigdeath";
     }
-    
+
     @Override
     protected int getDropItemId() {
         return Item.porkRaw.shiftedIndex;
     }
+
+    @Override
+    public Entity getBukkitEntity() {return this.lbPig;}
 }
