--- EntityPlayer.java
+++ EntityPlayer.java
@@ -1,4 +1,12 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.bukkit.event.player.PlayerDropItemEvent;
+import ru.vladthemountain.lilybukkit.core.entity.LBHumanEntity;
+
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
@@ -15,5 +23,7 @@
     private int damageRemainder;
     public Random playerRand;
-    
+
+    private final LBHumanEntity lbHumanEntity; //LilyBukkit
+
     public EntityPlayer(final World world) {
         super(world);
@@ -32,6 +42,8 @@
         this.texture = "/char.png";
         this.playerRand = new Random();
+
+        this.lbHumanEntity = new LBHumanEntity(Bukkit.getWorld(world.levelName), this); //LilyBukkit
     }
-    
+
     @Override
     public void updateRidden() {
@@ -40,5 +52,5 @@
         this.cameraYaw = 0.0f;
     }
-    
+
     @Override
     protected void updateEntityActionState() {
@@ -49,11 +61,10 @@
                 this.isSwinging = false;
             }
-        }
-        else {
+        } else {
             this.swingProgressInt = 0;
         }
         this.swingProgress = this.swingProgressInt / 8.0f;
     }
-    
+
     @Override
     public void onLivingUpdate() {
@@ -65,5 +76,5 @@
         super.onLivingUpdate();
         float f1 = MathHelper.sqrt_double(this.motionX * this.motionX + this.motionZ * this.motionZ);
-        float f2 = (float)Math.atan(-this.motionY * 0.20000000298023224) * 15.0f;
+        float f2 = (float) Math.atan(-this.motionY * 0.20000000298023224) * 15.0f;
         if (f1 > 0.1f) {
             f1 = 0.1f;
@@ -81,5 +92,5 @@
             if (list3 != null) {
                 for (int i4 = 0; i4 < list3.size(); ++i4) {
-                    this.collideWithPlayer((Entity)list3.get(i4));
+                    this.collideWithPlayer((Entity) list3.get(i4));
                 }
             }
@@ -90,10 +101,9 @@
             int n3 = 0;
             while (n2 != 0) {
-                final int n4 = this.worldObj.getBlockId((int)this.posX, n2, (int)this.posZ);
+                final int n4 = this.worldObj.getBlockId((int) this.posX, n2, (int) this.posZ);
                 if (n4 == 0 || n4 == 116) {
                     --n2;
                     n3 = 1;
-                }
-                else {
+                } else {
                     if (n3 != 0) {
                         ++n2;
@@ -105,12 +115,12 @@
             this.posY = n2;
             this.setPosition(this.posX, this.posY, this.posZ);
-            System.out.println(new StringBuilder("Sending player to Brazil at: ").append(this.posX).append(" ").append(this.posY).append(" ").append(this.posZ).toString());
+            System.out.print("Sending player to Brazil at: " + this.posX + " " + this.posY + " " + this.posZ);
         }
     }
-    
+
     private void collideWithPlayer(final Entity entity) {
         entity.onCollideWithPlayer(this);
     }
-    
+
     @Override
     public void onDeath(final Entity entity) {
@@ -121,10 +131,29 @@
             this.dropPlayerItemWithRandomChoice(new ItemStack(Item.appleRed, 1), true);
         }
-        this.inventory.dropAllItems();
+        // LilyBukkit start
+        List<org.bukkit.inventory.ItemStack> loot = new ArrayList<>();
+
+        for (int i = 0; i < this.inventory.getSizeInventory(); i++) {
+            ItemStack vanillaIS = this.inventory.getStackInSlot(i);
+            loot.add(new org.bukkit.inventory.ItemStack(vanillaIS.itemID, vanillaIS.stackSize, (short) vanillaIS.itemDmg));
+        }
+        //CraftBukkit start
+        org.bukkit.entity.Entity bukkitEntity = Bukkit.getPlayer(this.username);
+        org.bukkit.World bworld = Bukkit.getWorld(this.worldObj.levelName);
+
+        EntityDeathEvent event = new EntityDeathEvent(bukkitEntity, loot);
+        Bukkit.getServer().getPluginManager().callEvent(event);
+
+        for (org.bukkit.inventory.ItemStack stack : event.getDrops()) {
+            bworld.dropItemNaturally(bukkitEntity.getLocation(), stack);
+        }
+        //CraftBukkit end
+        this.inventory = new InventoryPlayer(this);
+        // LilyBukkit end
+        //this.inventory.dropAllItems();
         if (entity != null) {
             this.motionX = -MathHelper.cos((this.attackedAtYaw + this.rotationYaw) * 3.1415927f / 180.0f) * 0.1f;
             this.motionZ = -MathHelper.sin((this.attackedAtYaw + this.rotationYaw) * 3.1415927f / 180.0f) * 0.1f;
-        }
-        else {
+        } else {
             final double n = 0.0;
             this.motionZ = n;
@@ -133,14 +162,14 @@
         this.yOffset = 0.1f;
     }
-    
+
     @Override
     public void addToPlayerScore(final Entity entity, final int score) {
         this.score += score;
     }
-    
+
     public void dropPlayerItem(final ItemStack stack) {
         this.dropPlayerItemWithRandomChoice(stack, false);
     }
-    
+
     public void dropPlayerItemWithRandomChoice(final ItemStack stack, final boolean z2) {
         if (stack != null) {
@@ -154,6 +183,5 @@
                 entityItem3.motionZ = MathHelper.cos(f6) * f5;
                 entityItem3.motionY = 0.20000000298023224;
-            }
-            else {
+            } else {
                 f4 = 0.3f;
                 entityItem3.motionX = -MathHelper.sin(this.rotationYaw / 180.0f * 3.1415927f) * MathHelper.cos(this.rotationPitch / 180.0f * 3.1415927f) * f4;
@@ -163,19 +191,30 @@
                 final float f5 = this.rand.nextFloat() * 3.1415927f * 2.0f;
                 f4 *= this.rand.nextFloat();
-                final EntityItem entityItem4 = entityItem3;
-                entityItem4.motionX += Math.cos((double)f5) * f4;
-                final EntityItem entityItem5 = entityItem3;
-                entityItem5.motionY += (this.rand.nextFloat() - this.rand.nextFloat()) * 0.1f;
-                final EntityItem entityItem6 = entityItem3;
-                entityItem6.motionZ += Math.sin((double)f5) * f4;
+                entityItem3.motionX += Math.cos(f5) * f4;
+                entityItem3.motionY += (this.rand.nextFloat() - this.rand.nextFloat()) * 0.1f;
+                entityItem3.motionZ += Math.sin(f5) * f4;
             }
+
+            // CraftBukkit start
+            Player player = Bukkit.getPlayer(this.username);
+            org.bukkit.entity.Item drop = (org.bukkit.entity.Item) entityItem3.getBukkitEntity();
+
+            PlayerDropItemEvent event = new PlayerDropItemEvent(player, drop);
+            Bukkit.getServer().getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                player.getInventory().addItem(drop.getItemStack());
+                return;
+            }
+            // CraftBukkit end
+
             this.joinEntityItemWithWorld(entityItem3);
         }
     }
-    
+
     protected void joinEntityItemWithWorld(final EntityItem entityItem) {
-        this.worldObj.spawnEntityInWorld(entityItem);
+        this.worldObj.spawnEntityInWorld(entityItem, SpawnReason.NATURAL);
     }
-    
+
     public float getCurrentPlayerStrVsBlock(final Block block) {
         float f2 = this.inventory.getStrVsBlock(block);
@@ -188,9 +227,9 @@
         return f2;
     }
-    
+
     public boolean canHarvestBlock(final Block block) {
         return this.inventory.canHarvestBlock(block);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
@@ -199,25 +238,25 @@
         this.inventory.readFromNBT(nBTTagList2);
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
         super.writeEntityToNBT(nbttagcompound);
-        nbttagcompound.setTag("Inventory", (NBTBase)this.inventory.writeToNBT(new NBTTagList()));
+        nbttagcompound.setTag("Inventory", this.inventory.writeToNBT(new NBTTagList()));
     }
-    
+
     public void displayGUIChest(final IInventory inventory) {
     }
-    
+
     public void displayWorkbenchGUI() {
     }
-    
+
     public void onItemPickup(final Entity entity, final int i2) {
     }
-    
+
     @Override
     protected float getEyeHeight() {
         return 0.12f;
     }
-    
+
     @Override
     public boolean attackEntityFrom(final Entity entity, int damage) {
@@ -247,27 +286,32 @@
         return damage != 0 && super.attackEntityFrom(entity, damage);
     }
-    
+
     public void displayGUIFurnace(final TileEntityFurnace tileEntityFurnace) {
     }
-    
+
     public void displayGUIEditSign(final TileEntitySign tileEntitySign) {
     }
-    
+
     public ItemStack getCurrentEquippedItem() {
         return this.inventory.getCurrentItem();
     }
-    
+
     public void destroyCurrentEquippedItem() {
         this.inventory.setInventorySlotContents(this.inventory.currentItem, null);
     }
-    
+
     @Override
     public double getYOffset() {
         return this.yOffset - 0.5f;
     }
-    
+
     public void swingItem() {
         this.swingProgressInt = -1;
         this.isSwinging = true;
     }
+
+    @Override
+    public org.bukkit.entity.Entity getBukkitEntity() {
+        return this.lbHumanEntity;
+    }
 }
