--- EntitySheep.java
+++ EntitySheep.java
@@ -1,7 +1,14 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.event.entity.EntityDeathEvent;
+import ru.vladthemountain.lilybukkit.LBWorld;
+import ru.vladthemountain.lilybukkit.entity.LBSheep;
+
+import java.util.ArrayList;
+
 public class EntitySheep extends EntityAnimal {
     public boolean sheared;
-    
+
     public EntitySheep(final World world) {
         super(world);
@@ -10,5 +17,5 @@
         this.setSize(0.9f, 1.3f);
     }
-    
+
     @Override
     public boolean attackEntityFrom(final Entity entity, final int damage) {
@@ -27,5 +34,5 @@
         return super.attackEntityFrom(entity, damage);
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
@@ -33,5 +40,5 @@
         nbttagcompound.setBoolean("Sheared", this.sheared);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
@@ -39,18 +46,26 @@
         this.sheared = nbttagcompound.getBoolean("Sheared");
     }
-    
+
     @Override
     protected String getLivingSound() {
         return "mob.sheep";
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.sheep";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.sheep";
     }
+
+    @Override
+    public void onEntityDeath() {
+        super.onEntityDeath();
+        // LilyBukkit start
+        Bukkit.getPluginManager().callEvent(new EntityDeathEvent(new LBSheep((LBWorld) Bukkit.getWorld(this.worldObj.levelName), this), new ArrayList<>()));
+        // LilyBukkit end
+    }
 }
