--- EntitySheep.java
+++ EntitySheep.java
@@ -1,7 +1,10 @@
 package net.minecraft.src;
 
-public class EntitySheep extends EntityAnimal {
+import org.bukkit.DyeColor;
+import org.bukkit.entity.Sheep;
+
+public class EntitySheep extends EntityAnimal implements Sheep {
     public boolean sheared;
-    
+
     public EntitySheep(final World world) {
         super(world);
@@ -10,5 +13,5 @@
         this.setSize(0.9f, 1.3f);
     }
-    
+
     @Override
     public boolean attackEntityFrom(final Entity entity, final int damage) {
@@ -27,5 +30,5 @@
         return super.attackEntityFrom(entity, damage);
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
@@ -33,5 +36,5 @@
         nbttagcompound.setBoolean("Sheared", this.sheared);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
@@ -39,18 +42,56 @@
         this.sheared = nbttagcompound.getBoolean("Sheared");
     }
-    
+
     @Override
     protected String getLivingSound() {
         return "mob.sheep";
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.sheep";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.sheep";
     }
+
+    /**
+     * @return Whether the sheep is sheared.
+     * @author Celtic Minstrel
+     */
+    @Override
+    public boolean isSheared() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * @param flag Whether to shear the sheep
+     * @author Celtic Minstrel
+     */
+    @Override
+    public void setSheared(boolean flag) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the color of this object.
+     *
+     * @return The DyeColor of this object.
+     */
+    @Override
+    public DyeColor getColor() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Sets the color of this object to the specified DyeColor.
+     *
+     * @param color The color of the object, as a DyeColor.
+     */
+    @Override
+    public void setColor(DyeColor color) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
 }
