--- EntitySpider.java
+++ EntitySpider.java
@@ -1,5 +1,7 @@
 package net.minecraft.src;
 
-public class EntitySpider extends EntityMob {
+import org.bukkit.entity.Spider;
+
+public class EntitySpider extends EntityMob implements Spider {
     public EntitySpider(final World world) {
         super(world);
@@ -8,9 +10,9 @@
         this.moveSpeed = 0.8f;
     }
-    
+
     public double getMountedYOffset() {
         return this.height * 0.75 - 0.5;
     }
-    
+
     @Override
     protected Entity findPlayerToAttack() {
@@ -22,20 +24,20 @@
         return null;
     }
-    
+
     @Override
     protected String getLivingSound() {
         return "mob.spider";
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.spider";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.spiderdeath";
     }
-    
+
     @Override
     protected void attackEntity(final Entity entity, final float damage) {
@@ -43,6 +45,5 @@
         if (f3 > 0.5f && this.rand.nextInt(100) == 0) {
             this.entityToAttack = null;
-        }
-        else if (damage > 2.0f && damage < 6.0f && this.rand.nextInt(10) == 0) {
+        } else if (damage > 2.0f && damage < 6.0f && this.rand.nextInt(10) == 0) {
             if (this.onGround) {
                 final double d4 = entity.posX - this.posX;
@@ -53,20 +54,19 @@
                 this.motionY = 0.4000000059604645;
             }
-        }
-        else {
+        } else {
             super.attackEntity(entity, damage);
         }
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
         super.writeEntityToNBT(nbttagcompound);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
         super.readEntityFromNBT(nbttagcompound);
     }
-    
+
     @Override
     protected int getDropItemId() {
