--- EntityPig.java
+++ EntityPig.java
@@ -1,7 +1,9 @@
 package net.minecraft.src;
 
-public class EntityPig extends EntityAnimal {
+import org.bukkit.entity.Pig;
+
+public class EntityPig extends EntityAnimal implements Pig {
     public boolean saddled;
-    
+
     public EntityPig(final World world) {
         super(world);
@@ -11,5 +13,5 @@
         this.saddled = false;
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
@@ -17,5 +19,5 @@
         nbttagcompound.setBoolean("Saddle", this.saddled);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
@@ -23,23 +25,43 @@
         this.saddled = nbttagcompound.getBoolean("Saddle");
     }
-    
+
     @Override
     protected String getLivingSound() {
         return "mob.pig";
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.pig";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.pigdeath";
     }
-    
+
     @Override
     protected int getDropItemId() {
         return Item.porkRaw.shiftedIndex;
     }
+
+    /**
+     * Check if the pig has a saddle.
+     *
+     * @return if the pig has been saddled.
+     */
+    @Override
+    public boolean hasSaddle() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Sets if the pig has a saddle or not
+     *
+     * @param saddled set if the pig has a saddle or not.
+     */
+    @Override
+    public void setSaddle(boolean saddled) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
 }
