--- EntityCreeper.java
+++ EntityCreeper.java
@@ -1,4 +1,8 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+import ru.vladthemountain.lilybukkit.entity.LBCreeper;
+
 public class EntityCreeper extends EntityMob {
     int timeSinceIgnited;
@@ -6,5 +10,7 @@
     int fuseDuration;
     int creeperState;
-    
+
+    private final LBCreeper lbCreeper; //LilyBukkit
+
     public EntityCreeper(final World world) {
         super(world);
@@ -12,16 +18,18 @@
         this.creeperState = -1;
         this.texture = "/mob/creeper.png";
+
+        this.lbCreeper = new LBCreeper(Bukkit.getWorld(world.levelName), this); //LilyBukkit
     }
-    
+
     @Override
     public void writeEntityToNBT(final NBTTagCompound nbttagcompound) {
         super.writeEntityToNBT(nbttagcompound);
     }
-    
+
     @Override
     public void readEntityFromNBT(final NBTTagCompound nbttagcompound) {
         super.readEntityFromNBT(nbttagcompound);
     }
-    
+
     @Override
     protected void updateEntityActionState() {
@@ -38,15 +46,15 @@
         }
     }
-    
+
     @Override
     protected String getHurtSound() {
         return "mob.creeper";
     }
-    
+
     @Override
     protected String getDeathSound() {
         return "mob.creeperdeath";
     }
-    
+
     @Override
     public void onDeath(final Entity entity) {
@@ -56,24 +64,36 @@
         }
     }
-    
+
     @Override
     protected void attackEntity(final Entity entity, final float damage) {
         if ((this.creeperState <= 0 && damage < 3.0f) || (this.creeperState > 0 && damage < 7.0f)) {
             if (this.timeSinceIgnited == 0) {
-                this.worldObj.playSoundAtEntity((Entity)this, "random.fuse", 1.0f, 0.5f);
+                this.worldObj.playSoundAtEntity(this, "random.fuse", 1.0f, 0.5f);
             }
             this.creeperState = 1;
             ++this.timeSinceIgnited;
             if (this.timeSinceIgnited == this.fuseDuration) {
-                this.worldObj.createExplosion(this, this.posX, this.posY, this.posZ, 3.0f);
-                this.setEntityDead();
+                // CraftBukkit start
+                ExplosionPrimeEvent event = new ExplosionPrimeEvent(this.getBukkitEntity(), 3.0f, false);
+                Bukkit.getServer().getPluginManager().callEvent(event);
+
+                if (!event.isCancelled()) {
+                    this.worldObj.createExplosion(this, this.posX, this.posY, this.posZ, 3.0f);
+                    this.setEntityDead();
+                } else {
+                    this.timeSinceIgnited = 0;
+                }
+                // CraftBukkit end
             }
             this.hasAttacked = true;
         }
     }
-    
+
     @Override
     protected int getDropItemId() {
         return Item.gunpowder.shiftedIndex;
     }
+
+    @Override
+    public org.bukkit.entity.Entity getBukkitEntity() {return this.lbCreeper;}
 }
