--- IChunkProvider.java
+++ IChunkProvider.java
@@ -3,13 +3,13 @@
 public interface IChunkProvider {
     boolean chunkExists(final int x, final int z);
-    
+
     Chunk provideChunk(final int x, final int z);
-    
+
     void populate(final IChunkProvider chunkProvider, final int x, final int z);
-    
+
     boolean saveChunks(final boolean flag, final IProgressUpdate progressUpdate);
-    
+
     boolean unload100OldestChunks();
-    
+
     boolean canSave();
 }
