--- IChunkLoader.java
+++ IChunkLoader.java
@@ -5,11 +5,11 @@
 public interface IChunkLoader {
     Chunk loadChunk(final World world, final int x, final int z) throws IOException;
-    
+
     void saveChunk(final World world, final Chunk chunk) throws IOException;
-    
+
     void saveExtraChunkData(final World world, final Chunk chunk);
-    
+
     void chunkTick();
-    
+
     void saveExtraData();
 }
