--- ItemBlock.java
+++ ItemBlock.java
@@ -3,5 +3,5 @@
 public class ItemBlock extends Item {
     private int blockID;
-    
+
     public ItemBlock(final int itemID) {
         super(itemID);
@@ -9,11 +9,10 @@
         this.setIconIndex(Block.blocksList[itemID + 256].getBlockTextureFromSide(2));
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, int x, int y, int z, int i7) {
         if (world.getBlockId(x, y, z) == Block.snow.blockID) {
             i7 = 0;
-        }
-        else {
+        } else {
             if (i7 == 0) {
                 --y;
