--- ItemTool.java
+++ ItemTool.java
@@ -2,9 +2,9 @@
 
 public class ItemTool extends Item {
+    protected int toolMaterial;
     private Block[] blocksEffectiveAgainst;
     private float efficiencyOnProperMaterial;
     private int damageVsEntity;
-    protected int toolMaterial;
-    
+
     public ItemTool(final int itemID, final int damage, final int toolMaterial, final Block[] effectiveBlocks) {
         super(itemID);
@@ -17,8 +17,8 @@
             this.maxDamage *= 4;
         }
-        this.efficiencyOnProperMaterial = (float)((toolMaterial + 1) * 2);
+        this.efficiencyOnProperMaterial = (float) ((toolMaterial + 1) * 2);
         this.damageVsEntity = damage + toolMaterial;
     }
-    
+
     @Override
     public float getStrVsBlock(final ItemStack stack, final Block block) {
@@ -30,5 +30,5 @@
         return 1.0f;
     }
-    
+
     @Override
     public void onBlockDestroyed(final ItemStack stack, final int x, final int y, final int z, final int i5) {
