--- ItemDoor.java
+++ ItemDoor.java
@@ -1,4 +1,10 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.block.BlockPlaceEvent;
+import ru.vladthemountain.lilybukkit.LBWorld;
+import ru.vladthemountain.lilybukkit.block.LBBlockState;
+
 public class ItemDoor extends Item {
     private Material material;
@@ -16,4 +22,6 @@
             return false;
         }
+
+        int clickedX = x, clickedY = y, clickedZ = z; // CraftBukkit
         ++y;
         Block block8;
@@ -42,6 +50,6 @@
             b10 = 1;
         }
-        final int i9 = (world.isBlockNormalCube(x - b10, y, z - b11) + world.isBlockNormalCube(x - b10, y + 1, z - b11)) ? 1 : 0;
-        final int i10 = (world.isBlockNormalCube(x + b10, y, z + b11) + world.isBlockNormalCube(x + b10, y + 1, z + b11)) ? 1 : 0;
+final int i9 = (world.isBlockNormalCube(x - b10, y, z - b11) && world.isBlockNormalCube(x - b10, y + 1, z - b11)) ? 1 : 0;
+final int i10 = (world.isBlockNormalCube(x + b10, y, z + b11) && world.isBlockNormalCube(x + b10, y + 1, z + b11)) ? 1 : 0;
         final boolean z2 = world.getBlockId(x - b10, y, z - b11) == block8.blockID || world.getBlockId(x - b10, y + 1, z - b11) == block8.blockID;
         final boolean z3 = world.getBlockId(x + b10, y, z + b11) == block8.blockID || world.getBlockId(x + b10, y + 1, z + b11) == block8.blockID;
@@ -57,6 +65,17 @@
             i8 += 4;
         }
+
+        LBBlockState blockState = new LBBlockState((LBWorld) Bukkit.getWorld(world.levelName), Bukkit.getWorld(world.levelName).getBlockAt(x,y,z)); //LilyBukkit
+
         world.setBlockWithNotify(x, y, z, block8.blockID);
         world.setBlockMetadataWithNotify(x, y, z, i8);
+        //CraftBukkit start
+        BlockPlaceEvent event = CraftEventFactory.callBlockPlaceEvent(world, (EntityPlayerMP) entityPlayer, blockState, clickedX, clickedY, clickedZ, this.material==Material.wood?Block.doorWood:Block.doorSteel);
+
+        if (event.isCancelled() || !event.canBuild()) {
+            event.getBlockPlaced().setTypeIdAndData(blockState.getTypeId(), blockState.getRawData(), false);
+            return false;
+        }
+        //CraftBukkit end
         world.setBlockWithNotify(x, y + 1, z, block8.blockID);
         world.setBlockMetadataWithNotify(x, y + 1, z, i8 + 8);
