--- ItemDoor.java
+++ ItemDoor.java
@@ -3,5 +3,5 @@
 public class ItemDoor extends Item {
     private Material material;
-    
+
     public ItemDoor(final int itemID, final Material material) {
         super(itemID);
@@ -10,5 +10,5 @@
         this.maxStackSize = 1;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, int y, final int z, final int i7) {
@@ -20,6 +20,5 @@
         if (this.material == Material.wood) {
             block8 = Block.doorWood;
-        }
-        else {
+        } else {
             block8 = Block.doorSteel;
         }
@@ -49,6 +48,5 @@
         if (z2 && !z3) {
             z4 = true;
-        }
-        else if (i10 > i9) {
+        } else if (i10 > i9) {
             z4 = true;
         }
