--- ItemSign.java
+++ ItemSign.java
@@ -7,5 +7,5 @@
         this.maxStackSize = 1;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, int x, int y, int z, final int i7) {
@@ -36,10 +36,9 @@
         if (i7 == 1) {
             world.setBlockAndMetadataWithNotify(x, y, z, Block.signStanding.blockID, MathHelper.floor_double((entityPlayer.rotationYaw + 180.0f) * 16.0f / 360.0f + 0.5) & 0xF);
-        }
-        else {
+        } else {
             world.setBlockAndMetadataWithNotify(x, y, z, Block.signWall.blockID, i7);
         }
         --stack.stackSize;
-        entityPlayer.displayGUIEditSign((TileEntitySign)world.getBlockTileEntity(x, y, z));
+        entityPlayer.displayGUIEditSign((TileEntitySign) world.getBlockTileEntity(x, y, z));
         return true;
     }
