--- ItemSeeds.java
+++ ItemSeeds.java
@@ -3,10 +3,10 @@
 public class ItemSeeds extends Item {
     private int blockType;
-    
+
     public ItemSeeds(final int itemID, final int blockType) {
         super(itemID);
         this.blockType = blockType;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, final int y, final int z, final int i7) {
