--- ItemSeeds.java
+++ ItemSeeds.java
@@ -1,12 +1,18 @@
 package net.minecraft.src;
 
+import org.bukkit.Bukkit;
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.block.BlockPlaceEvent;
+import ru.vladthemountain.lilybukkit.core.LBWorld;
+import ru.vladthemountain.lilybukkit.core.block.LBBlockState;
+
 public class ItemSeeds extends Item {
     private int blockType;
-    
+
     public ItemSeeds(final int itemID, final int blockType) {
         super(itemID);
         this.blockType = blockType;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, final int y, final int z, final int i7) {
@@ -15,6 +21,18 @@
         }
         final int i8 = world.getBlockId(x, y, z);
-        if (i8 == Block.tilledField.blockID) {
+        if (i8 == Block.TILLED_FIELD.blockID) {
+            LBBlockState blockState = new LBBlockState((LBWorld) Bukkit.getWorld(world.levelName), Bukkit.getWorld(world.levelName).getBlockAt(x, y, z)); //LilyBukkit
+
             world.setBlockWithNotify(x, y + 1, z, this.blockType);
+
+            // CraftBukkit start - seeds
+            BlockPlaceEvent event = CraftEventFactory.callBlockPlaceEvent(world, (EntityPlayerMP) entityPlayer, blockState, x, y, z, this.blockType);
+
+            if (event.isCancelled() || !event.canBuild()) {
+                event.getBlockPlaced().setTypeId(0);
+                return false;
+            }
+            // CraftBukkit end
+
             --stack.stackSize;
             return true;
