--- ItemStack.java
+++ ItemStack.java
@@ -6,25 +6,25 @@
     public int itemID;
     public int itemDmg;
-    
+
     public ItemStack(final Block block) {
         this(block, 1);
     }
-    
+
     public ItemStack(final Block block, final int count) {
         this(block.blockID, count);
     }
-    
+
     public ItemStack(final Item item) {
         this(item, 1);
     }
-    
+
     public ItemStack(final Item item, final int count) {
         this(item.shiftedIndex, count);
     }
-    
+
     public ItemStack(final int itemID) {
         this(itemID, 1);
     }
-    
+
     public ItemStack(final int itemID, final int stackSize) {
         this.stackSize = 0;
@@ -32,5 +32,5 @@
         this.stackSize = stackSize;
     }
-    
+
     public ItemStack(final int itemID, final int stackSize, final int itemDmg) {
         this.stackSize = 0;
@@ -39,29 +39,29 @@
         this.itemDmg = itemDmg;
     }
-    
+
     public ItemStack(final NBTTagCompound nbttagcompound) {
         this.stackSize = 0;
         this.readFromNBT(nbttagcompound);
     }
-    
+
     public Item getItem() {
         return Item.itemsList[this.itemID];
     }
-    
+
     public boolean useItem(final EntityPlayer entityPlayer, final World world, final int x, final int y, final int z, final int i6) {
         return this.getItem().onItemUse(this, entityPlayer, world, x, y, z, i6);
     }
-    
+
     public float getStrVsBlock(final Block block) {
         return this.getItem().getStrVsBlock(this, block);
     }
-    
+
     public NBTTagCompound writeToNBT(final NBTTagCompound nbttagcompound) {
-        nbttagcompound.setShort("id", (short)this.itemID);
-        nbttagcompound.setByte("Count", (byte)this.stackSize);
-        nbttagcompound.setShort("Damage", (short)this.itemDmg);
+        nbttagcompound.setShort("id", (short) this.itemID);
+        nbttagcompound.setByte("Count", (byte) this.stackSize);
+        nbttagcompound.setShort("Damage", (short) this.itemDmg);
         return nbttagcompound;
     }
-    
+
     public void readFromNBT(final NBTTagCompound nbttagcompound) {
         this.itemID = nbttagcompound.getShort("id");
@@ -69,13 +69,13 @@
         this.itemDmg = nbttagcompound.getShort("Damage");
     }
-    
+
     public int getMaxStackSize() {
         return this.getItem().getItemStackLimit();
     }
-    
+
     public int getMaxDamage() {
         return Item.itemsList[this.itemID].getMaxDamage();
     }
-    
+
     public void damageItem(final int damage) {
         this.itemDmg += damage;
@@ -88,16 +88,16 @@
         }
     }
-    
+
     public void onDestroyBlock(final int x, final int y, final int z, final int i4) {
         Item.itemsList[this.itemID].onBlockDestroyed(this, x, y, z, i4);
     }
-    
+
     public boolean canHarvestBlock(final Block block) {
         return Item.itemsList[this.itemID].canHarvestBlock(block);
     }
-    
+
     public void onItemDestroyedByUse(final EntityPlayer entityPlayer) {
     }
-    
+
     public ItemStack copy() {
         return new ItemStack(this.itemID, this.stackSize, this.itemDmg);
