--- Item.java
+++ Item.java
@@ -4,5 +4,4 @@
 
 public class Item {
-    protected static Random rand;
     public static Item[] itemsList;
     public static Item shovel;
@@ -98,9 +97,4 @@
     public static Item record13;
     public static Item recordCat;
-    public final int shiftedIndex;
-    protected int maxStackSize;
-    protected int maxDamage;
-    protected int iconIndex;
-    protected boolean bFull3D;
     public static Item obsidianArmor0;
     public static Item obsidianArmor1;
@@ -120,5 +114,6 @@
     public static Item edibleFire;
     public static Item flameberge;
-    
+    protected static Random rand;
+
     static {
         Item.rand = new Random();
@@ -234,5 +229,11 @@
         Item.flameberge = new ItemSwordFlameberge(106).setIconIndex(69);
     }
-    
+
+    public final int shiftedIndex;
+    protected int maxStackSize;
+    protected int maxDamage;
+    protected int iconIndex;
+    protected boolean bFull3D;
+
     protected Item(final int itemID) {
         this.maxStackSize = 64;
@@ -245,37 +246,37 @@
         Item.itemsList[256 + itemID] = this;
     }
-    
+
     public Item setIconIndex(final int iconIndex) {
         this.iconIndex = iconIndex;
         return this;
     }
-    
+
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, final int y, final int z, final int i7) {
         return false;
     }
-    
+
     public float getStrVsBlock(final ItemStack stack, final Block block) {
         return 1.0f;
     }
-    
+
     public ItemStack onItemRightClick(final ItemStack itemStack, final World world, final EntityPlayer entityPlayer) {
         return itemStack;
     }
-    
+
     public int getItemStackLimit() {
         return this.maxStackSize;
     }
-    
+
     public int getMaxDamage() {
         return this.maxDamage;
     }
-    
+
     public void onBlockDestroyed(final ItemStack stack, final int x, final int y, final int z, final int i5) {
     }
-    
+
     public boolean canHarvestBlock(final Block block) {
         return false;
     }
-    
+
     public Item setFull3D() {
         this.bFull3D = true;
