--- ItemPickaxe.java
+++ ItemPickaxe.java
@@ -3,15 +3,16 @@
 public class ItemPickaxe extends ItemTool {
     private static Block[] blocksEffectiveAgainst;
-    private int harvestLevel;
-    
+
     static {
-        ItemPickaxe.blocksEffectiveAgainst = new Block[] { Block.cobblestone, Block.stairDouble, Block.stairSingle, Block.stone, Block.cobblestoneMossy, Block.oreIron, Block.blockSteel, Block.oreCoal, Block.blockGold, Block.oreGold, Block.oreDiamond, Block.blockDiamond, Block.ice };
+        ItemPickaxe.blocksEffectiveAgainst = new Block[]{Block.cobblestone, Block.stairDouble, Block.stairSingle, Block.stone, Block.cobblestoneMossy, Block.oreIron, Block.blockSteel, Block.oreCoal, Block.blockGold, Block.oreGold, Block.oreDiamond, Block.blockDiamond, Block.ice};
     }
-    
+
+    private int harvestLevel;
+
     public ItemPickaxe(final int itemID, final int harvestLevelAndMaterial) {
         super(itemID, 2, harvestLevelAndMaterial, ItemPickaxe.blocksEffectiveAgainst);
         this.harvestLevel = harvestLevelAndMaterial;
     }
-    
+
     @Override
     public boolean canHarvestBlock(final Block block) {
