--- ItemRecord.java
+++ ItemRecord.java
@@ -3,5 +3,5 @@
 public class ItemRecord extends Item {
     private String recordName;
-    
+
     protected ItemRecord(final int itemID, final String recordName) {
         super(itemID);
@@ -9,5 +9,5 @@
         this.maxStackSize = 1;
     }
-    
+
     @Override
     public boolean onItemUse(final ItemStack stack, final EntityPlayer entityPlayer, final World world, final int x, final int y, final int z, final int i7) {
