--- MapGenBase.java
+++ MapGenBase.java
@@ -6,10 +6,10 @@
     protected int range;
     protected Random rand;
-    
+
     public MapGenBase() {
         this.range = 8;
         this.rand = new Random();
     }
-    
+
     public void generate(final ChunkProviderGenerate chunkProviderGenerate, final World world, final int i3, final int i4, final byte[] data) {
         final int i5 = this.range;
@@ -24,5 +24,5 @@
         }
     }
-    
+
     protected void recursiveGenerate(final World world, final int i2, final int i3, final int i4, final int i5, final byte[] data) {
     }
