--- MapGenCaves.java
+++ MapGenCaves.java
@@ -7,5 +7,5 @@
         this.generateCaveNode(i1, i2, data, d4, d6, d8, 1.0f + this.rand.nextFloat() * 6.0f, 0.0f, 0.0f, -1, -1, 0.5);
     }
-    
+
     protected void generateCaveNode(final int x, final int z, final byte[] data, double d4, double d6, double d8, final float f10, float f11, float f12, int i13, int i14, final double d15) {
         final double d16 = x * 16 + 8;
@@ -35,6 +35,5 @@
             if (z3) {
                 f12 *= 0.92f;
-            }
-            else {
+            } else {
                 f12 *= 0.7f;
             }
@@ -115,10 +114,9 @@
                                         if (b51 == Block.stone.blockID || b51 == Block.dirt.blockID || b51 == Block.grass.blockID) {
                                             if (i28 < 10) {
-                                                data[i27] = (byte)Block.lavaMoving.blockID;
-                                            }
-                                            else {
+                                                data[i27] = (byte) Block.lavaMoving.blockID;
+                                            } else {
                                                 data[i27] = 0;
                                                 if (z5 && data[i27 - 1] == Block.dirt.blockID) {
-                                                    data[i27 - 1] = (byte)Block.grass.blockID;
+                                                    data[i27 - 1] = (byte) Block.grass.blockID;
                                                 }
                                             }
@@ -138,5 +136,5 @@
         }
     }
-    
+
     @Override
     protected void recursiveGenerate(final World world, final int i2, final int i3, final int i4, final int i5, final byte[] data) {
