--- MCHashEntry2.java
+++ MCHashEntry2.java
@@ -3,8 +3,8 @@
 class MCHashEntry2 {
     final long hashEntry;
+    final int slotHash;
     Object valueEntry;
     MCHashEntry2 nextEntry;
-    final int slotHash;
-    
+
     MCHashEntry2(final int slotHash, final long hashEntry, final Object valueEntry, final MCHashEntry2 nextEntry) {
         this.valueEntry = valueEntry;
@@ -13,18 +13,18 @@
         this.slotHash = slotHash;
     }
-    
+
     public final long getHash() {
         return this.hashEntry;
     }
-    
+
     public final Object getValue() {
         return this.valueEntry;
     }
-    
+
     public final boolean equals(final Object object) {
         if (!(object instanceof MCHashEntry2)) {
             return false;
         }
-        final MCHashEntry2 mCHashEntry22 = (MCHashEntry2)object;
+        final MCHashEntry2 mCHashEntry22 = (MCHashEntry2) object;
         final Long long3 = this.getHash();
         final Long long4 = mCHashEntry22.getHash();
@@ -38,9 +38,9 @@
         return false;
     }
-    
+
     public final int hashCode() {
         return MCHashTable2.getHash(this.hashEntry);
     }
-    
+
     public final String toString() {
         return String.valueOf(this.getHash()) + "=" + this.getValue();
