--- NBTTagString.java
+++ NBTTagString.java
@@ -2,13 +2,13 @@
 
 import java.io.DataInput;
-import java.io.IOException;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public class NBTTagString extends NBTBase {
     public String stringValue;
-    
+
     public NBTTagString() {
     }
-    
+
     public NBTTagString(final String stringValue) {
         this.stringValue = stringValue;
@@ -17,20 +17,20 @@
         }
     }
-    
+
     @Override
     void writeTagContents(final DataOutput dataOutput) throws IOException {
         dataOutput.writeUTF(this.stringValue);
     }
-    
+
     @Override
     void readTagContents(final DataInput dataInput) throws IOException {
         this.stringValue = dataInput.readUTF();
     }
-    
+
     @Override
     public byte getType() {
         return 8;
     }
-    
+
     public String toString() {
         return new StringBuilder().append(this.stringValue).toString();
