--- NBTTagLong.java
+++ NBTTagLong.java
@@ -2,32 +2,32 @@
 
 import java.io.DataInput;
-import java.io.IOException;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public class NBTTagLong extends NBTBase {
     public long longValue;
-    
+
     public NBTTagLong() {
     }
-    
+
     public NBTTagLong(final long longValue) {
         this.longValue = longValue;
     }
-    
+
     @Override
     void writeTagContents(final DataOutput dataOutput) throws IOException {
         dataOutput.writeLong(this.longValue);
     }
-    
+
     @Override
     void readTagContents(final DataInput dataInput) throws IOException {
         this.longValue = dataInput.readLong();
     }
-    
+
     @Override
     public byte getType() {
         return 4;
     }
-    
+
     public String toString() {
         return new StringBuilder().append(this.longValue).toString();
