--- NBTTagInt.java
+++ NBTTagInt.java
@@ -2,32 +2,32 @@
 
 import java.io.DataInput;
-import java.io.IOException;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public class NBTTagInt extends NBTBase {
     public int intValue;
-    
+
     public NBTTagInt() {
     }
-    
+
     public NBTTagInt(final int intValue) {
         this.intValue = intValue;
     }
-    
+
     @Override
     void writeTagContents(final DataOutput dataOutput) throws IOException {
         dataOutput.writeInt(this.intValue);
     }
-    
+
     @Override
     void readTagContents(final DataInput dataInput) throws IOException {
         this.intValue = dataInput.readInt();
     }
-    
+
     @Override
     public byte getType() {
         return 3;
     }
-    
+
     public String toString() {
         return new StringBuilder().append(this.intValue).toString();
