--- NBTTagFloat.java
+++ NBTTagFloat.java
@@ -2,32 +2,32 @@
 
 import java.io.DataInput;
-import java.io.IOException;
 import java.io.DataOutput;
+import java.io.IOException;
 
 public class NBTTagFloat extends NBTBase {
     public float floatValue;
-    
+
     public NBTTagFloat() {
     }
-    
+
     public NBTTagFloat(final float floatValue) {
         this.floatValue = floatValue;
     }
-    
+
     @Override
     void writeTagContents(final DataOutput dataOutput) throws IOException {
         dataOutput.writeFloat(this.floatValue);
     }
-    
+
     @Override
     void readTagContents(final DataInput dataInput) throws IOException {
         this.floatValue = dataInput.readFloat();
     }
-    
+
     @Override
     public byte getType() {
         return 5;
     }
-    
+
     public String toString() {
         return new StringBuilder().append(this.floatValue).toString();
