--- NBTTagEnd.java
+++ NBTTagEnd.java
@@ -1,7 +1,7 @@
 package net.minecraft.src;
 
+import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
-import java.io.DataInput;
 
 public class NBTTagEnd extends NBTBase {
@@ -9,14 +9,14 @@
     void readTagContents(final DataInput dataInput) throws IOException {
     }
-    
+
     @Override
     void writeTagContents(final DataOutput dataOutput) throws IOException {
     }
-    
+
     @Override
     public byte getType() {
         return 0;
     }
-    
+
     public String toString() {
         return "END";
