--- NetHandler.java
+++ NetHandler.java
@@ -4,107 +4,107 @@
     public void handleMapChunk(final Packet51MapChunk packet) {
     }
-    
+
     public void registerPacket(final Packet packet) {
     }
-    
+
     public void handleErrorMessage(final String message) {
     }
-    
+
     public void handleKickDisconnect(final Packet255KickDisconnect packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleLogin(final Packet1Login packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleFlying(final Packet10Flying packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleMultiBlockChange(final Packet52MultiBlockChange packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleBlockDig(final Packet14BlockDig packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleBlockChange(final Packet53BlockChange packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handlePreChunk(final Packet50PreChunk packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleNamedEntitySpawn(final Packet20NamedEntitySpawn packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleEntity(final Packet30Entity packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleEntityTeleport(final Packet34EntityTeleport packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handlePlace(final Packet15Place packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleBlockItemSwitch(final Packet16BlockItemSwitch packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleDestroyEntity(final Packet29DestroyEntity packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handlePickupSpawn(final Packet21PickupSpawn packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleCollect(final Packet22Collect packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleChat(final Packet3Chat packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleAddToInventory(final Packet17AddToInventory packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleVehicleSpawn(final Packet23VehicleSpawn packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleArmAnimation(final Packet18ArmAnimation packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleHandshake(final Packet2Handshake packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleMobSpawn(final Packet24MobSpawn packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleUpdateTime(final Packet4UpdateTime packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handlePlayerInventory(final Packet5PlayerInventory packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleComplexEntity(final Packet59ComplexEntity packet) {
         this.registerPacket(packet);
     }
-    
+
     public void handleSpawnPosition(final Packet6SpawnPosition packet) {
         this.registerPacket(packet);
