--- NetLoginHandler.java
+++ NetLoginHandler.java
@@ -1,9 +1,13 @@
 package net.minecraft.src;
 
-import java.net.HttpURLConnection;
-import java.net.URL;
+import net.minecraft.server.MinecraftServer;
+import org.bukkit.Bukkit;
+import org.bukkit.event.player.PlayerPreLoginEvent;
+
 import java.io.IOException;
+import java.net.HttpURLConnection;
 import java.net.Socket;
-import net.minecraft.server.MinecraftServer;
+import java.net.URL;
+import java.net.URLEncoder;
 import java.util.Random;
 import java.util.logging.Logger;
@@ -17,49 +21,57 @@
     private long loginStart;
     private String username;
+    private String password;
     private Packet1Login packet1login;
     private String serverId;
-    private boolean isULPP;
     private int ULPPVersion;
-    private boolean isULPPOnline;
-    private static final long loginTimeout = 30000L;
-    
+    public int ROSEVersion;
+    private static final long LOGIN_TIMEOUT = 30000L;
+
     static {
         NetLoginHandler.logger = Logger.getLogger("Minecraft");
         NetLoginHandler.rand = new Random();
     }
-    
+
     public NetLoginHandler(final MinecraftServer minecraftServer, final Socket socket, final String threadName) throws IOException {
         this.finishedProcessing = false;
         this.loginStart = System.currentTimeMillis();
         this.username = null;
+        this.password = "";
         this.packet1login = null;
         this.serverId = "";
-        this.isULPP = false;
         this.ULPPVersion = 0;
-        this.isULPPOnline = false;
+        this.ROSEVersion = 0;
         this.mcServer = minecraftServer;
         this.netManager = new NetworkManager(socket, threadName, this);
     }
-    
-    public void tryLogin() throws IOException {
+
+    public void tryLogin() {
         if (this.packet1login != null) {
+            // LilyBukkit start
+            PlayerPreLoginEvent event = new PlayerPreLoginEvent(this.packet1login.username, this.netManager.networkSocket.getInetAddress());
+            Bukkit.getServer().getPluginManager().callEvent(event);
+
+            if (event.getResult() != PlayerPreLoginEvent.Result.ALLOWED) {
+                this.kickUser(event.getKickMessage());
+                return;
+            }
+            // LilyBukkit end
             this.doLogin(this.packet1login);
             this.packet1login = null;
         }
-        if (this.loginStart + 30000L < System.currentTimeMillis()) {
+        if (this.loginStart + LOGIN_TIMEOUT < System.currentTimeMillis()) {
             this.kickUser("Took too long to log in");
-        }
-        else {
+        } else {
             this.netManager.processReadPackets();
         }
     }
-    
+
     public void kickUser(final String playerName) {
-        NetLoginHandler.logger.info(new StringBuilder("Disconnecting ").append(this.getUserAndIPString()).append(": ").append(playerName).toString());
+        NetLoginHandler.logger.info("Disconnecting " + this.getUserAndIPString() + ": " + playerName);
         this.netManager.addToSendQueue(new Packet255KickDisconnect(playerName));
         this.netManager.serverShutdown();
         this.finishedProcessing = true;
     }
-    
+
     @Override
     public void handleHandshake(final Packet2Handshake packet) {
@@ -67,63 +79,99 @@
             this.serverId = Long.toHexString(NetLoginHandler.rand.nextLong());
             this.netManager.addToSendQueue(new Packet2Handshake(this.serverId));
-        }
-        else {
+        } else {
             this.netManager.addToSendQueue(new Packet2Handshake("-"));
         }
     }
-    
+
     @Override
     public void handleLogin(final Packet1Login packet) {
         this.username = packet.username;
-        if (packet.protocolVersion != 2) {
+        this.password = packet.password;
+        if (packet.protocolVersion /* > */ != 2 /* || packet.protocolVersion < 1 */) {// :skull:
             this.kickUser("Outdated client!");
         }
-        this.isULPP = ULPPManager.IsVersionString(packet.password);
-        if (this.isULPP) {
-            this.ULPPVersion = ULPPManager.GetVersion(packet.password);
-            this.isULPPOnline = ULPPManager.IsOnline(packet.password);
+        if (ULPPManager.supportsExtension(packet.password, "ULPP", 1, 1)) {
+            NetLoginHandler.logger.info("User has ULPP enabled"); //LilyBukkit - additional output
+            this.ULPPVersion = 1;
+        } else {
+            NetLoginHandler.logger.info("User doesn't have ULPP enabled"); //LilyBukkit - additional output
         }
+        if (ULPPManager.supportsExtension(packet.password, "ROSE", 1, 1)) {
+            NetLoginHandler.logger.info("User connected via Rosepad 1.1.0.2-1.1.0.5"); //LilyBukkit - additional output
+            this.ROSEVersion = 1;
+        } else if (ULPPManager.supportsExtension(packet.password, "ROSE", 2, 2)) {
+            NetLoginHandler.logger.info("User connected via Rosepad 1.1.0.6+"); //LilyBukkit - additional output
+            this.ROSEVersion = 2;
+        } else {
+            NetLoginHandler.logger.info("User isn't using Rosepad"); //LilyBukkit - additional output
+        }
+        // LilyBukkit start
+        PlayerPreLoginEvent event = new PlayerPreLoginEvent(this.packet1login.username, this.netManager.networkSocket.getInetAddress());
+        Bukkit.getServer().getPluginManager().callEvent(event);
+
+        if (event.getResult() != PlayerPreLoginEvent.Result.ALLOWED) {
+            this.kickUser(event.getKickMessage());
+            return;
+        }
+        // LilyBukkit end
         this.doLogin(packet);
     }
-    
+
     @Override
     public void handleULPPLoggedIn(final Packet129ULPPLoggedIn packet) {
+        if (this.username == null) {
+            NetLoginHandler.logger.info("'unknown' failed to authenticate");
+            this.kickUser("Unable to authenticate using ULPP: no username found."); //LilyBukkit - altered the reason
+            return;
+        }
+        //LilyBukkit start
+        if (this.ULPPVersion == 0) {
+            NetLoginHandler.logger.info(this.username + " didn't provide a valid ULPP version, continuing without the ULPP login.");
+            return;
+        }
+        // LilyBukkit end
+        else if (this.ULPPVersion != 1) {
+            NetLoginHandler.logger.info(this.username + " failed to authenticate");
+            this.kickUser("You're using a newer ULPP protocol version. I'm still running ULPP1."); //LilyBukkit - changed the reason
+            return;
+        }
         try {
-            final URL checkURL = new URL(new StringBuilder("https://sessionserver.mojang.com/session/minecraft/hasJoined?username=").append(this.username).append("&serverId=").append(this.mcServer.serverId).toString());
-            final HttpURLConnection conn = (HttpURLConnection)checkURL.openConnection();
+            final URL checkURL = new URL("https://sessionserver.mojang.com/session/minecraft/hasJoined?username="
+                    + URLEncoder.encode(this.username, "utf-8") + "&serverId=" + this.mcServer.serverId);
+            final HttpURLConnection conn = (HttpURLConnection) checkURL.openConnection();
             conn.connect();
             final int returnCode = conn.getResponseCode();
             if (returnCode != 200) {
-                NetLoginHandler.logger.info(String.valueOf(this.username) + " failed to authenticate");
-                this.kickUser("Unable to authenticate using ULPP");
+                NetLoginHandler.logger.info(this.username + " failed to authenticate");
+                this.kickUser("Unable to authenticate using ULPP. HTTP returned code: " + returnCode + "."); //LilyBukkit - add returnCode to the reason
                 return;
             }
-        }
-        catch (IOException e) {
+        } catch (IOException e) {
             e.printStackTrace();
-            NetLoginHandler.logger.info(String.valueOf(this.username) + " failed to authenticate");
-            this.kickUser("Unable to authenticate using ULPP");
+            NetLoginHandler.logger.info(this.username + " failed to authenticate");
+            this.kickUser("Unable to authenticate using ULPP. Caught an IOException: " + e.getLocalizedMessage()); //LilyBukkit - add Exception's message
             return;
         }
-        NetLoginHandler.logger.info(String.valueOf(this.username) + " authenticated using ULPP");
-        final EntityPlayerMP loggedInPlayer = this.mcServer.configManager.login(this, this.username, "", true);
+        final EntityPlayerMP loggedInPlayer = this.mcServer.configManager.login(this, this.username, this.password, true);
         if (loggedInPlayer == null) {
             this.finishedProcessing = true;
+            return;
         }
         final int minVersion = Integer.min(this.ULPPVersion, 1);
         final NetServerHandler ULPPLoginServerHandler = ULPPManager.CreateServerHandler(minVersion, this.mcServer, this.netManager, loggedInPlayer);
-        ULPPLoginServerHandler.sendPacket(new Packet6SpawnPosition(this.mcServer.worldMngr.spawnX, this.mcServer.worldMngr.spawnY, this.mcServer.worldMngr.spawnZ));
+        ULPPLoginServerHandler.sendPacket(new Packet6SpawnPosition(this.mcServer.getWorldServer(loggedInPlayer.worldObj.levelName).spawnX, this.mcServer.getWorldServer(loggedInPlayer.worldObj.levelName).spawnY, this.mcServer.getWorldServer(loggedInPlayer.worldObj.levelName).spawnZ));
         this.mcServer.configManager.playerLoggedIn(loggedInPlayer);
         ULPPLoginServerHandler.teleportTo(loggedInPlayer.posX, loggedInPlayer.posY, loggedInPlayer.posZ, loggedInPlayer.rotationYaw, loggedInPlayer.rotationPitch);
         ULPPLoginServerHandler.sendInventoryPackets();
         this.mcServer.networkServer.addPlayer(ULPPLoginServerHandler);
-        ULPPLoginServerHandler.sendPacket(new Packet4UpdateTime(this.mcServer.worldMngr.worldTime));
+        ULPPLoginServerHandler.sendPacket(new Packet4UpdateTime(this.mcServer.getWorldServer(loggedInPlayer.worldObj.levelName).worldTime));
+        NetLoginHandler.logger.info(this.username + " authenticated using ULPP");
         this.finishedProcessing = true;
     }
-    
+
     public void doLogin(final Packet1Login loginPacket) {
         final int minVersion = Integer.min(this.ULPPVersion, 1);
         final int usedULPPVer = 1;
-        if (this.isULPPOnline && minVersion > 0 && this.mcServer.ULPPOnlineMode) {
+        if (this.ULPPVersion == 1 && this.mcServer.ULPPOnlineMode) {
             this.netManager.addToSendQueue(new Packet1Login("", this.mcServer.serverId, usedULPPVer));
             return;
@@ -131,42 +179,41 @@
         final EntityPlayerMP entityPlayerMP2 = this.mcServer.configManager.login(this, loginPacket.username, loginPacket.password, false);
         if (entityPlayerMP2 != null) {
-            NetLoginHandler.logger.info(String.valueOf(this.getUserAndIPString()) + " logged in with ULPP" + Integer.toString(this.ULPPVersion));
+            NetLoginHandler.logger.info(this.getUserAndIPString() + " logged in with ULPP" + this.ULPPVersion);
             NetServerHandler netServerHandler3;
             if (minVersion == 0) {
                 netServerHandler3 = new NetServerHandler(this.mcServer, this.netManager, entityPlayerMP2);
-            }
-            else {
+            } else {
                 netServerHandler3 = ULPPManager.CreateServerHandler(minVersion, this.mcServer, this.netManager, entityPlayerMP2);
             }
             netServerHandler3.sendPacket(new Packet1Login("", "", usedULPPVer));
-            netServerHandler3.sendPacket(new Packet6SpawnPosition(this.mcServer.worldMngr.spawnX, this.mcServer.worldMngr.spawnY, this.mcServer.worldMngr.spawnZ));
+            netServerHandler3.sendPacket(new Packet6SpawnPosition(this.mcServer.getWorldServer(entityPlayerMP2.worldObj.levelName).spawnX, this.mcServer.getWorldServer(entityPlayerMP2.worldObj.levelName).spawnY, this.mcServer.getWorldServer(entityPlayerMP2.worldObj.levelName).spawnZ));
             this.mcServer.configManager.playerLoggedIn(entityPlayerMP2);
             netServerHandler3.teleportTo(entityPlayerMP2.posX, entityPlayerMP2.posY, entityPlayerMP2.posZ, entityPlayerMP2.rotationYaw, entityPlayerMP2.rotationPitch);
             netServerHandler3.sendInventoryPackets();
             this.mcServer.networkServer.addPlayer(netServerHandler3);
-            netServerHandler3.sendPacket(new Packet4UpdateTime(this.mcServer.worldMngr.worldTime));
+            netServerHandler3.sendPacket(new Packet4UpdateTime(this.mcServer.getWorldServer(entityPlayerMP2.worldObj.levelName).worldTime));
         }
         this.finishedProcessing = true;
     }
-    
+
     @Override
     public void handleErrorMessage(final String message) {
-        NetLoginHandler.logger.info(String.valueOf(this.getUserAndIPString()) + " lost connection");
+        NetLoginHandler.logger.info(this.getUserAndIPString() + " lost connection");
         this.finishedProcessing = true;
     }
-    
+
     @Override
     public void registerPacket(final Packet packet) {
         this.kickUser("Protocol error");
     }
-    
+
     public String getUserAndIPString() {
-        return (this.username != null) ? (String.valueOf(this.username) + " [" + this.netManager.getRemoteAddress().toString() + "]") : this.netManager.getRemoteAddress().toString();
+        return (this.username != null) ? (this.username + " [" + this.netManager.getRemoteAddress().toString() + "]") : this.netManager.getRemoteAddress().toString();
     }
-    
+
     static String getServerId(final NetLoginHandler loginHandler) {
         return loginHandler.serverId;
     }
-    
+
     static Packet1Login setLoginPacket(final NetLoginHandler loginHandler, final Packet1Login loginPacket) {
         return loginHandler.packet1login = loginPacket;
