--- NetworkWriterThread.java
+++ NetworkWriterThread.java
@@ -3,10 +3,10 @@
 class NetworkWriterThread extends Thread {
     final NetworkManager netManager;
-    
+
     NetworkWriterThread(final NetworkManager netManager, final String threadName) {
         super(threadName);
         this.netManager = netManager;
     }
-    
+
     public void run() {
         Object object1 = NetworkManager.threadSyncObject;
@@ -23,6 +23,5 @@
                 }
                 NetworkManager.sendNetworkPacket(this.netManager);
-            }
-            finally {
+            } finally {
                 if (z11) {
                     final Object object2 = NetworkManager.threadSyncObject;
@@ -40,10 +39,10 @@
                     continue;
                 }
-                object1 = NetworkManager.threadSyncObject;
+                /* object1 = NetworkManager.threadSyncObject;
                 synchronized (NetworkManager.threadSyncObject) {
                     --NetworkManager.numWriteThreads;
                 }
                 // monitorexit(NetworkManager.threadSyncObject)
-                break;
+                break;*/
             }
         }
