--- NibbleArray.java
+++ NibbleArray.java
@@ -3,13 +3,13 @@
 public class NibbleArray {
     public final byte[] data;
-    
+
     public NibbleArray(final int data) {
         this.data = new byte[data >> 1];
     }
-    
+
     public NibbleArray(final byte[] data) {
         this.data = data;
     }
-    
+
     public int get(final int i1, final int i2, final int i3) {
         final int i4 = i1 << 11 | i3 << 7 | i2;
@@ -18,5 +18,5 @@
         return (i6 == 0) ? (this.data[i5] & 0xF) : (this.data[i5] >> 4 & 0xF);
     }
-    
+
     public void set(final int i1, final int i2, final int i3, final int i4) {
         final int i5 = i1 << 11 | i3 << 7 | i2;
@@ -24,11 +24,10 @@
         final int i7 = i5 & 0x1;
         if (i7 == 0) {
-            this.data[i6] = (byte)((this.data[i6] & 0xF0) | (i4 & 0xF));
+            this.data[i6] = (byte) ((this.data[i6] & 0xF0) | (i4 & 0xF));
+        } else {
+            this.data[i6] = (byte) ((this.data[i6] & 0xF) | (i4 & 0xF) << 4);
         }
-        else {
-            this.data[i6] = (byte)((this.data[i6] & 0xF) | (i4 & 0xF) << 4);
-        }
     }
-    
+
     public boolean isValid() {
         return this.data != null;
