--- ObsidianPick.java
+++ ObsidianPick.java
@@ -1,21 +1,18 @@
 package net.minecraft.src;
 
-import net.minecraft.src.Material;
-import net.minecraft.src.Block;
-import net.minecraft.src.ItemTool;
-
 public class ObsidianPick extends ItemTool {
     private static Block[] blocksEffectiveAgainst;
-    private int harvestLevel;
-    
+
     static {
-        ObsidianPick.blocksEffectiveAgainst = new Block[] { Block.cobblestone, Block.bedrock, Block.stairDouble, Block.stairSingle, Block.stone, Block.cobblestoneMossy, Block.oreIron, Block.blockSteel, Block.oreCoal, Block.blockGold, Block.oreGold, Block.oreDiamond, Block.blockDiamond, Block.ice };
+        ObsidianPick.blocksEffectiveAgainst = new Block[]{Block.cobblestone, Block.bedrock, Block.stairDouble, Block.stairSingle, Block.stone, Block.cobblestoneMossy, Block.oreIron, Block.blockSteel, Block.oreCoal, Block.blockGold, Block.oreGold, Block.oreDiamond, Block.blockDiamond, Block.ice};
     }
-    
+
+    private int harvestLevel;
+
     public ObsidianPick(final int n, final int n2) {
         super(n, 2, n2, ObsidianPick.blocksEffectiveAgainst);
         this.harvestLevel = n2;
     }
-    
+
     @Override
     public boolean canHarvestBlock(final Block block) {
