--- Packet22Collect.java
+++ Packet22Collect.java
@@ -1,20 +1,20 @@
 package net.minecraft.src;
 
+import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
-import java.io.DataInputStream;
 
 public class Packet22Collect extends Packet {
     public int collectedEntityId;
     public int collectorEntityId;
-    
+
     public Packet22Collect() {
     }
-    
+
     public Packet22Collect(final int collectedEntityID, final int collectorEntityID) {
         this.collectedEntityId = collectedEntityID;
         this.collectorEntityId = collectorEntityID;
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
@@ -22,5 +22,5 @@
         this.collectorEntityId = dataInputStream.readInt();
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
@@ -28,10 +28,10 @@
         dataOutputStream.writeInt(this.collectorEntityId);
     }
-    
+
     @Override
     public void processPacket(final NetHandler netHandler) {
         netHandler.handleCollect(this);
     }
-    
+
     @Override
     public int getPacketSize() {
