--- Packet10Flying.java
+++ Packet10Flying.java
@@ -1,7 +1,7 @@
 package net.minecraft.src;
 
+import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
-import java.io.DataInputStream;
 
 public class Packet10Flying extends Packet {
@@ -15,20 +15,20 @@
     public boolean moving;
     public boolean rotating;
-    
+
     @Override
     public void processPacket(final NetHandler netHandler) {
         netHandler.handleFlying(this);
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
         this.onGround = (dataInputStream.read() != 0);
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
-        dataOutputStream.write((int)(this.onGround ? 1 : 0));
+        dataOutputStream.write((int) (this.onGround ? 1 : 0));
     }
-    
+
     @Override
     public int getPacketSize() {
