--- Packet51MapChunk.java
+++ Packet51MapChunk.java
@@ -1,10 +1,10 @@
 package net.minecraft.src;
 
+import java.io.DataInputStream;
 import java.io.DataOutputStream;
-import java.util.zip.DataFormatException;
 import java.io.IOException;
-import java.util.zip.Inflater;
-import java.io.DataInputStream;
+import java.util.zip.DataFormatException;
 import java.util.zip.Deflater;
+import java.util.zip.Inflater;
 
 public class Packet51MapChunk extends Packet {
@@ -17,9 +17,9 @@
     public byte[] chunkData;
     private int tempLength;
-    
+
     public Packet51MapChunk() {
         this.isChunkDataPacket = true;
     }
-    
+
     public Packet51MapChunk(final int xPosition, final int yPosition, final int zPosition, final int xSize, final int ySize, final int zSize, final World world) {
         this.isChunkDataPacket = true;
@@ -37,11 +37,10 @@
             this.chunkData = new byte[xSize * ySize * zSize * 5 / 2];
             this.tempLength = deflater9.deflate(this.chunkData);
-        }
-        finally {
+        } finally {
             deflater9.end();
         }
         deflater9.end();
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
@@ -60,14 +59,12 @@
         try {
             inflater4.inflate(this.chunkData);
-        }
-        catch (DataFormatException dataFormatException9) {
+        } catch (DataFormatException dataFormatException9) {
             throw new IOException("Bad compressed data format");
-        }
-        finally {
+        } finally {
             inflater4.end();
         }
         inflater4.end();
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
@@ -81,10 +78,10 @@
         dataOutputStream.write(this.chunkData, 0, this.tempLength);
     }
-    
+
     @Override
     public void processPacket(final NetHandler netHandler) {
         netHandler.handleMapChunk(this);
     }
-    
+
     @Override
     public int getPacketSize() {
