--- Packet30Entity.java
+++ Packet30Entity.java
@@ -1,7 +1,7 @@
 package net.minecraft.src;
 
+import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
-import java.io.DataInputStream;
 
 public class Packet30Entity extends Packet {
@@ -13,29 +13,29 @@
     public byte pitch;
     public boolean rotating;
-    
+
     public Packet30Entity() {
         this.rotating = false;
     }
-    
+
     public Packet30Entity(final int entityID) {
         this.rotating = false;
         this.entityId = entityID;
     }
-    
+
     @Override
     public void readPacketData(final DataInputStream dataInputStream) throws IOException {
         this.entityId = dataInputStream.readInt();
     }
-    
+
     @Override
     public void writePacket(final DataOutputStream dataOutputStream) throws IOException {
         dataOutputStream.writeInt(this.entityId);
     }
-    
+
     @Override
     public void processPacket(final NetHandler netHandler) {
         netHandler.handleEntity(this);
     }
-    
+
     @Override
     public int getPacketSize() {
