--- Pathfinder.java
+++ Pathfinder.java
@@ -6,5 +6,5 @@
     private MCHashTable pointMap;
     private PathPoint[] pathOptions;
-    
+
     public Pathfinder(final IBlockAccess worldMap) {
         this.path = new Path();
@@ -13,13 +13,13 @@
         this.worldMap = worldMap;
     }
-    
+
     public PathEntity createEntityPathTo(final Entity entity1, final Entity entity2, final float f3) {
         return this.createEntityPathTo(entity1, entity2.posX, entity2.boundingBox.minY, entity2.posZ, f3);
     }
-    
+
     public PathEntity createEntityPathTo(final Entity entity, final int x, final int boundingBoxY, final int z, final float f5) {
         return this.createEntityPathTo(entity, x + 0.5f, boundingBoxY + 0.5f, z + 0.5f, f5);
     }
-    
+
     private PathEntity createEntityPathTo(final Entity entity, final double x, final double boundingBoxY, final double z, final float f8) {
         this.path.clearPath();
@@ -31,5 +31,5 @@
         return pathEntity12;
     }
-    
+
     private PathEntity addToPath(final Entity entity, final PathPoint pathPoint1, final PathPoint pathPoint2, final PathPoint pathPoint3, final float f5) {
         pathPoint1.totalPathDistance = 0.0f;
@@ -57,6 +57,5 @@
                     if (pathPoint6.isAssigned()) {
                         this.path.changeDistance(pathPoint6, pathPoint6.totalPathDistance + pathPoint6.distanceToNext);
-                    }
-                    else {
+                    } else {
                         pathPoint6.distanceToTarget = pathPoint6.totalPathDistance + pathPoint6.distanceToNext;
                         this.path.addPoint(pathPoint6);
@@ -70,5 +69,5 @@
         return this.createEntityPath(pathPoint1, pathPoint4);
     }
-    
+
     private int findPathOptions(final Entity entity, final PathPoint pathPoint1, final PathPoint pathPoint2, final PathPoint pathPoint3, final float f5) {
         int i6 = 0;
@@ -95,5 +94,5 @@
         return i6;
     }
-    
+
     private PathPoint getSafePoint(final Entity entity, final int i2, int i3, final int i4, final PathPoint pathPoint, final int i6) {
         PathPoint pathPoint2 = null;
@@ -124,8 +123,8 @@
         return pathPoint2;
     }
-    
+
     private final PathPoint openPoint(final int x, final int y, final int z) {
         final int i4 = x | y << 10 | z << 20;
-        PathPoint pathPoint5 = (PathPoint)this.pointMap.lookup(i4);
+        PathPoint pathPoint5 = (PathPoint) this.pointMap.lookup(i4);
         if (pathPoint5 == null) {
             pathPoint5 = new PathPoint(x, y, z);
@@ -134,5 +133,5 @@
         return pathPoint5;
     }
-    
+
     private int getVerticalOffset(final Entity entity, final int x, final int y, final int z, final PathPoint pathPoint) {
         for (int i6 = x; i6 < x + pathPoint.xCoord; ++i6) {
@@ -151,5 +150,5 @@
         return 1;
     }
-    
+
     private PathEntity createEntityPath(final PathPoint pathPoint1, final PathPoint pathPoint2) {
         int i3 = 1;
