--- PlayerManager.java
+++ PlayerManager.java
@@ -4,22 +4,23 @@
 import java.util.ArrayList;
 import net.minecraft.server.MinecraftServer;
+
 import java.util.List;
 
 public class PlayerManager {
-    private List players;
+    private List<EntityPlayerMP> players;
     private MCHashTable2 playerInstances;
-    private List playerInstancesToUpdate;
+    private List<PlayerInstance> playerInstancesToUpdate;
     private MinecraftServer mcServer;
     
     public PlayerManager(final MinecraftServer mcServer) {
-        this.players = (List)new ArrayList();
+        this.players = new ArrayList<>();
         this.playerInstances = new MCHashTable2();
-        this.playerInstancesToUpdate = (List)new ArrayList();
+        this.playerInstancesToUpdate = new ArrayList<>();
         this.mcServer = mcServer;
     }
     
     public void updatePlayerInstances() throws IOException {
-        for (int i1 = 0; i1 < this.playerInstancesToUpdate.size(); ++i1) {
-            ((PlayerInstance)this.playerInstancesToUpdate.get(i1)).onUpdate();
+        for (PlayerInstance playerInstance : this.playerInstancesToUpdate) {
+            playerInstance.onUpdate();
         }
         this.playerInstancesToUpdate.clear();
@@ -55,5 +56,5 @@
     
     public void addPlayer(final EntityPlayerMP entityPlayerMP) {
-        this.mcServer.configManager.sendPacketToAllPlayers(new Packet3Chat(new StringBuilder("§e").append(entityPlayerMP.username).append(" joined the game.").toString()));
+        //this.mcServer.configManager.sendPacketToAllPlayers(new Packet3Chat("§e" + entityPlayerMP.username + " joined the game."));
         final int i2 = (int)entityPlayerMP.posX >> 4;
         final int i3 = (int)entityPlayerMP.posZ >> 4;
@@ -69,5 +70,5 @@
     
     public void removePlayer(final EntityPlayerMP entityPlayerMP) {
-        this.mcServer.configManager.sendPacketToAllPlayers(new Packet3Chat(new StringBuilder("§e").append(entityPlayerMP.username).append(" left the game.").toString()));
+        this.mcServer.configManager.sendPacketToAllPlayers(new Packet3Chat("§e" +entityPlayerMP.username + " left the game.")); //LilyBukkit: replace 'this.playerEntity.username' with own code
         final int i2 = (int)entityPlayerMP.posX >> 4;
         final int i3 = (int)entityPlayerMP.posZ >> 4;
@@ -132,5 +133,5 @@
     }
     
-    static List getPlayerInstancesToUpdate(final PlayerManager playerManager) {
+    static List<PlayerInstance> getPlayerInstancesToUpdate(final PlayerManager playerManager) {
         return playerManager.playerInstancesToUpdate;
     }
