--- PlayerManager.java
+++ PlayerManager.java
@@ -4,4 +4,8 @@
 import java.util.ArrayList;
 import net.minecraft.server.MinecraftServer;
+import org.bukkit.event.player.PlayerJoinEvent;
+import ru.vladthemountain.lilybukkit.LBWorld;
+import ru.vladthemountain.lilybukkit.entity.LBPlayer;
+
 import java.util.List;
 
@@ -66,4 +70,6 @@
         }
         this.players.add(entityPlayerMP);
+        // LilyBukkit
+        this.mcServer.lilyBukkit.getPluginManager().callEvent(new PlayerJoinEvent(new LBPlayer((LBWorld) this.mcServer.lilyBukkit.getWorld(this.mcServer.worldMngr.levelName), entityPlayerMP), "§a" + entityPlayerMP.username + " joined the server"));
     }
     
