--- PlayerListBox.java
+++ PlayerListBox.java
@@ -1,22 +1,23 @@
 package net.minecraft.src;
 
-import java.util.Vector;
 import net.minecraft.server.MinecraftServer;
-import javax.swing.JList;
 
+import javax.swing.*;
+import java.util.Vector;
+
 public class PlayerListBox extends JList implements IUpdatePlayerListBox {
     private MinecraftServer mcServer;
     private int playerCount;
-    
+
     public PlayerListBox(final MinecraftServer minecraftServer) {
         this.playerCount = 0;
         (this.mcServer = minecraftServer).addToOnlinePlayerList(this);
     }
-    
+
     public void addAllPlayers() {
         if (this.playerCount++ % 20 == 0) {
             final Vector vector1 = new Vector();
             for (int i2 = 0; i2 < this.mcServer.configManager.playerEntities.size(); ++i2) {
-                vector1.add(((EntityPlayerMP)this.mcServer.configManager.playerEntities.get(i2)).username);
+                vector1.add(((EntityPlayerMP) this.mcServer.configManager.playerEntities.get(i2)).username);
             }
             this.setListData(vector1);
