--- PlayerNBTManager.java
+++ PlayerNBTManager.java
@@ -1,23 +1,20 @@
 package net.minecraft.src;
 
-import java.io.InputStream;
-import java.io.FileInputStream;
-import java.io.OutputStream;
-import java.io.FileOutputStream;
-import java.io.File;
+import java.io.*;
 import java.util.logging.Logger;
 
 public class PlayerNBTManager {
     public static Logger logger;
-    private File playerNBT;
-    
+
     static {
         PlayerNBTManager.logger = Logger.getLogger("Minecraft");
     }
-    
+
+    private File playerNBT;
+
     public PlayerNBTManager(final File playerNBT) {
         (this.playerNBT = playerNBT).mkdir();
     }
-    
+
     public void writePlayerNBT(final EntityPlayerMP entityPlayerMP) {
         try {
@@ -26,26 +23,24 @@
             final File file3 = new File(this.playerNBT, "_tmp_.dat");
             final File file4 = new File(this.playerNBT, String.valueOf(entityPlayerMP.username) + ".dat");
-            CompressedStreamTools.writeCompressed(nBTTagCompound2, (OutputStream)new FileOutputStream(file3));
+            CompressedStreamTools.writeCompressed(nBTTagCompound2, (OutputStream) new FileOutputStream(file3));
             if (file4.exists()) {
                 file4.delete();
             }
             file3.renameTo(file4);
-        }
-        catch (Exception exception5) {
+        } catch (Exception exception5) {
             PlayerNBTManager.logger.warning(new StringBuilder("Failed to save player data for ").append(entityPlayerMP.username).toString());
         }
     }
-    
+
     public void readPlayerNBT(final EntityPlayerMP entityPlayerMP) {
         try {
             final File file2 = new File(this.playerNBT, String.valueOf(entityPlayerMP.username) + ".dat");
             if (file2.exists()) {
-                final NBTTagCompound nBTTagCompound3 = CompressedStreamTools.readCompressed((InputStream)new FileInputStream(file2));
+                final NBTTagCompound nBTTagCompound3 = CompressedStreamTools.readCompressed((InputStream) new FileInputStream(file2));
                 if (nBTTagCompound3 != null) {
                     entityPlayerMP.readFromNBT(nBTTagCompound3);
                 }
             }
-        }
-        catch (Exception exception4) {
+        } catch (Exception exception4) {
             PlayerNBTManager.logger.warning(new StringBuilder("Failed to load player data for ").append(entityPlayerMP.username).toString());
         }
