--- SkyGen.java
+++ SkyGen.java
@@ -2,14 +2,12 @@
 
 import java.util.Random;
-import net.minecraft.src.World;
-import net.minecraft.src.WorldGenerator;
 
 public class SkyGen extends WorldGenerator {
     private int blockID;
-    
+
     public SkyGen(final int n) {
         this.blockID = n;
     }
-    
+
     @Override
     public boolean generate(final World world, final Random rand, final int x, final int y, final int z) {
@@ -21,6 +19,5 @@
             if (n5 > 90) {
                 bl = true;
-            }
-            else if (n5 > 80) {
+            } else if (n5 > 80) {
                 bl = (rand.nextInt(100) > 80);
             }
