--- SpawnerAnimals.java
+++ SpawnerAnimals.java
@@ -1,5 +1,4 @@
 package net.minecraft.src;
 
-import java.util.Iterator;
 import java.util.HashSet;
 import java.util.Set;
@@ -9,5 +8,5 @@
     private Class entityType;
     private Class[] entities;
-    private Set eligibleChunksForSpawning;
+private Set<ChunkCoordIntPair> eligibleChunksForSpawning;
     
     public SpawnerAnimals(final int i1, final Class class2, final Class[] class3) {
