--- StepSound.java
+++ StepSound.java
@@ -5,5 +5,5 @@
     public final float stepSoundVolume;
     public final float stepSoundPitch;
-    
+
     public StepSound(final String string1, final float f2, final float f3) {
         this.stepSoundName = string1;
@@ -11,13 +11,13 @@
         this.stepSoundPitch = f3;
     }
-    
+
     public float getVolume() {
         return this.stepSoundVolume;
     }
-    
+
     public float getPitch() {
         return this.stepSoundPitch;
     }
-    
+
     public String getStepSound() {
         return new StringBuilder("step.").append(this.stepSoundName).toString();
