--- TileEntitySign.java
+++ TileEntitySign.java
@@ -4,10 +4,10 @@
     public String[] signText;
     public int lineBeingEdited;
-    
+
     public TileEntitySign() {
-        this.signText = new String[] { "", "", "", "" };
+        this.signText = new String[]{"", "", "", ""};
         this.lineBeingEdited = -1;
     }
-    
+
     @Override
     public void writeToNBT(final NBTTagCompound nbttagcompound) {
@@ -18,5 +18,5 @@
         nbttagcompound.setString("Text4", this.signText[3]);
     }
-    
+
     @Override
     public void readFromNBT(final NBTTagCompound nbttagcompound) {
