--- WorldGenFlowers.java
+++ WorldGenFlowers.java
@@ -5,9 +5,9 @@
 public class WorldGenFlowers extends WorldGenerator {
     private int plantBlockID;
-    
+
     public WorldGenFlowers(final int plantBlockID) {
         this.plantBlockID = plantBlockID;
     }
-    
+
     @Override
     public boolean generate(final World world, final Random rand, final int x, final int y, final int z) {
@@ -21,5 +21,5 @@
                         continue;
                     }
-                    if (!((BlockFlower)Block.blocksList[this.plantBlockID]).canBlockStay(world, n6, n7, n8)) {
+                    if (!((BlockFlower) Block.blocksList[this.plantBlockID]).canBlockStay(world, n6, n7, n8)) {
                         continue;
                     }
