--- World.java
+++ World.java
@@ -19,10 +19,10 @@
 
 public class World implements IBlockAccess {
-    private List lightingToUpdate;
-    public List loadedEntityList;
-    private List unloadedEntityList;
-    private TreeSet scheduledTickTreeSet;
-    private Set scheduledTickSet;
-    public List loadedTileEntityList;
+    private List<MetadataChunkBlock> lightingToUpdate;
+    public List<Entity> loadedEntityList;
+    private List<Entity> unloadedEntityList;
+    private TreeSet<NextTickListEntry> scheduledTickTreeSet;
+    private Set<NextTickListEntry> scheduledTickSet;
+    public List<TileEntity> loadedTileEntityList;
     public long worldTime;
     public boolean snowCovered;
@@ -37,5 +37,5 @@
     private final long lockTimestamp;
     protected int autosavePeriod;
-    public List playerEntities;
+    public List<Entity> playerEntities;
     public int difficultySetting;
     public Random rand;
@@ -44,5 +44,5 @@
     public int spawnZ;
     public boolean isNewWorld;
-    protected List worldAccesses;
+    protected List<IWorldAccess> worldAccesses;
     private IChunkProvider chunkProvider;
     public File saveDirectory;
@@ -52,6 +52,6 @@
     public final String levelName;
     public boolean worldChunkLoadOverride;
-    private ArrayList collidingBoundingBoxes;
-    private Set positionsToUpdate;
+    private ArrayList<AxisAlignedBB> collidingBoundingBoxes;
+private Set<ChunkCoordIntPair> positionsToUpdate;
     private int soundCounter;
     private List entitiesWithinAABBExcludingEntity;
