--- MinecraftServer.java
+++ MinecraftServer.java
@@ -1,43 +1,35 @@
 package net.minecraft.server;
 
-import net.minecraft.src.ThreadServerApplication;
-import net.minecraft.src.ServerGUI;
-import java.awt.GraphicsEnvironment;
-import net.minecraft.src.Packet3Chat;
-import net.minecraft.src.ItemStack;
-import net.minecraft.src.Item;
-import net.minecraft.src.EntityPlayerMP;
-import net.minecraft.src.ServerCommand;
-import java.util.Iterator;
-import net.minecraft.src.IUpdatePlayerListBox;
-import net.minecraft.src.Packet;
-import net.minecraft.src.Packet4UpdateTime;
-import net.minecraft.src.Vec3D;
-import net.minecraft.src.AxisAlignedBB;
-import net.minecraft.src.IProgressUpdate;
-import net.minecraft.src.IWorldAccess;
-import net.minecraft.src.WorldManager;
+import com.avaje.ebean.config.ServerConfig;
+import net.minecraft.src.*;
+import org.bukkit.Server;
+import org.bukkit.World;
+import org.bukkit.command.CommandException;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.PluginCommand;
+import org.bukkit.entity.Player;
+import org.bukkit.generator.ChunkGenerator;
+import org.bukkit.inventory.Recipe;
+import org.bukkit.map.MapView;
+import org.bukkit.plugin.PluginManager;
+import org.bukkit.plugin.ServicesManager;
+import org.bukkit.plugin.SimplePluginManager;
+import org.bukkit.plugin.SimpleServicesManager;
+import org.bukkit.scheduler.BukkitScheduler;
+import ru.vtm.lilybukkit.plugin.LBPluginManager;
+import ru.vtm.lilybukkit.server.LBServerProperties;
+
+import java.awt.*;
+import java.io.File;
 import java.io.IOException;
-import java.util.logging.Level;
 import java.net.InetAddress;
-import java.io.File;
-import net.minecraft.src.ConsoleLogManager;
-import net.minecraft.src.ThreadCommandReader;
-import net.minecraft.src.ThreadSleepForeverServer;
-import java.util.Collections;
-import java.util.ArrayList;
-import net.minecraft.src.EntityTracker;
 import java.util.List;
-import net.minecraft.src.ServerConfigurationManager;
-import net.minecraft.src.WorldServer;
-import net.minecraft.src.PropertyManager;
-import net.minecraft.src.NetworkListenThread;
-import java.util.HashMap;
+import java.util.*;
+import java.util.logging.Level;
 import java.util.logging.Logger;
-import net.minecraft.src.ICommandListener;
 
-public class MinecraftServer implements ICommandListener, Runnable {
+public class MinecraftServer implements ICommandListener, Runnable, Server {
     public static Logger logger;
-    public static HashMap playerList;
+    public static HashMap<String, Integer> playerList;
     public NetworkListenThread networkServer;
     public PropertyManager propertyManagerObj;
@@ -53,19 +45,19 @@
     public EntityTracker entityTracker;
     public boolean onlineMode;
-    
+
     static {
         MinecraftServer.logger = Logger.getLogger("Minecraft");
         MinecraftServer.playerList = new HashMap();
     }
-    
+
     public MinecraftServer() {
         this.serverRunning = true;
         this.serverStopped = false;
         this.deathTime = 0;
-        this.playersOnline = (List)new ArrayList();
-        this.commands = Collections.synchronizedList((List)new ArrayList());
+        this.playersOnline = (List) new ArrayList();
+        this.commands = Collections.synchronizedList((List) new ArrayList());
         new ThreadSleepForeverServer(this);
     }
-    
+
     private boolean startServer() throws IOException {
         final ThreadCommandReader threadCommandReader1 = new ThreadCommandReader(this);
@@ -90,6 +82,5 @@
         try {
             this.networkServer = new NetworkListenThread(this, inetAddress3, i4);
-        }
-        catch (IOException iOException6) {
+        } catch (IOException iOException6) {
             MinecraftServer.logger.warning("**** FAILED TO BIND TO PORT!");
             MinecraftServer.logger.log(Level.WARNING, new StringBuilder("The exception was: ").append(iOException6.toString()).toString());
@@ -111,5 +102,5 @@
         return true;
     }
-    
+
     private void initWorld(final String worldName) {
         MinecraftServer.logger.info("Preparing start region");
@@ -129,5 +120,5 @@
         this.clearCurrentTask();
     }
-    
+
     private void outputPercentRemaining(final String currentTask, final int percent) {
         this.currentTask = currentTask;
@@ -135,15 +126,15 @@
         System.out.println(String.valueOf(currentTask) + ": " + percent + "%");
     }
-    
+
     private void clearCurrentTask() {
         this.currentTask = null;
         this.percentDone = 0;
     }
-    
+
     private void save() {
         MinecraftServer.logger.info("Saving chunks");
         this.worldMngr.saveWorld(true, null);
     }
-    
+
     private void stop() {
         MinecraftServer.logger.info("Stopping server");
@@ -155,9 +146,9 @@
         }
     }
-    
+
     public void stopRunning() {
         this.serverRunning = false;
     }
-    
+
     public void run() {
         try {
@@ -184,32 +175,27 @@
                     Thread.sleep(1L);
                 }
-            }
-            else {
+            } else {
                 while (this.serverRunning) {
                     this.commandLineParser();
                     try {
                         Thread.sleep(10L);
-                    }
-                    catch (InterruptedException interruptedException15) {
+                    } catch (InterruptedException interruptedException15) {
                         interruptedException15.printStackTrace();
                     }
                 }
             }
-        }
-        catch (Exception exception16) {
+        } catch (Exception exception16) {
             exception16.printStackTrace();
-            MinecraftServer.logger.log(Level.SEVERE, "Unexpected exception", (Throwable)exception16);
+            MinecraftServer.logger.log(Level.SEVERE, "Unexpected exception", (Throwable) exception16);
             while (this.serverRunning) {
                 this.commandLineParser();
                 try {
                     Thread.sleep(10L);
-                }
-                catch (InterruptedException interruptedException16) {
+                } catch (InterruptedException interruptedException16) {
                     interruptedException16.printStackTrace();
                 }
             }
             return;
-        }
-        finally {
+        } finally {
             this.stop();
             this.serverStopped = true;
@@ -220,13 +206,12 @@
         System.exit(0);
     }
-    
+
     private void doTick() throws IOException {
         final ArrayList arrayList1 = new ArrayList();
         for (final String string3 : MinecraftServer.playerList.keySet()) {
-            final int i4 = (int)MinecraftServer.playerList.get(string3);
+            final int i4 = (int) MinecraftServer.playerList.get(string3);
             if (i4 > 0) {
                 MinecraftServer.playerList.put(string3, (i4 - 1));
-            }
-            else {
+            } else {
                 arrayList1.add(string3);
             }
@@ -242,5 +227,6 @@
         }
         this.worldMngr.tick();
-        while (this.worldMngr.updatingLighting()) {}
+        while (this.worldMngr.updatingLighting()) {
+        }
         this.worldMngr.updateEntities();
         this.networkServer.handleNetworkListenThread();
@@ -248,21 +234,20 @@
         this.entityTracker.updateTrackedEntities();
         for (int i5 = 0; i5 < this.playersOnline.size(); ++i5) {
-            ((IUpdatePlayerListBox)this.playersOnline.get(i5)).addAllPlayers();
+            ((IUpdatePlayerListBox) this.playersOnline.get(i5)).addAllPlayers();
         }
         try {
             this.commandLineParser();
+        } catch (Exception exception5) {
+            MinecraftServer.logger.log(Level.WARNING, "Unexpected exception while parsing console command", (Throwable) exception5);
         }
-        catch (Exception exception5) {
-            MinecraftServer.logger.log(Level.WARNING, "Unexpected exception while parsing console command", (Throwable)exception5);
-        }
     }
-    
+
     public void addCommand(final String command, final ICommandListener commandListener) {
         this.commands.add(new ServerCommand(command, commandListener));
     }
-    
+
     public void commandLineParser() {
         while (this.commands.size() > 0) {
-            final ServerCommand serverCommand1 = (ServerCommand)this.commands.remove(0);
+            final ServerCommand serverCommand1 = (ServerCommand) this.commands.remove(0);
             String string2 = serverCommand1.command;
             final ICommandListener iCommandListener3 = serverCommand1.commandListener;
@@ -271,45 +256,36 @@
                 if (string2.toLowerCase().startsWith("list")) {
                     iCommandListener3.addHelpCommandMessage(new StringBuilder("Connected players: ").append(this.configManager.getPlayerList()).toString());
-                }
-                else if (string2.toLowerCase().startsWith("stop")) {
+                } else if (string2.toLowerCase().startsWith("stop")) {
                     this.print(string3, "Stopping the server..");
                     this.serverRunning = false;
-                }
-                else if (string2.toLowerCase().startsWith("save-all")) {
+                } else if (string2.toLowerCase().startsWith("save-all")) {
                     this.print(string3, "Forcing save..");
                     this.worldMngr.saveWorld(true, null);
                     this.print(string3, "Save complete.");
-                }
-                else if (string2.toLowerCase().startsWith("save-off")) {
+                } else if (string2.toLowerCase().startsWith("save-off")) {
                     this.print(string3, "Disabling level saving..");
                     this.worldMngr.levelSaving = true;
-                }
-                else if (string2.toLowerCase().startsWith("save-on")) {
+                } else if (string2.toLowerCase().startsWith("save-on")) {
                     this.print(string3, "Enabling level saving..");
                     this.worldMngr.levelSaving = false;
-                }
-                else if (string2.toLowerCase().startsWith("op ")) {
+                } else if (string2.toLowerCase().startsWith("op ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     this.configManager.opPlayer(string4);
                     this.print(string3, new StringBuilder("Opping ").append(string4).toString());
                     this.configManager.sendChatMessageToPlayer(string4, "§eYou are now op!");
-                }
-                else if (string2.toLowerCase().startsWith("deop ")) {
+                } else if (string2.toLowerCase().startsWith("deop ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     this.configManager.deopPlayer(string4);
                     this.configManager.sendChatMessageToPlayer(string4, "§eYou are no longer op!");
                     this.print(string3, new StringBuilder("De-opping ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("ban-ip ")) {
+                } else if (string2.toLowerCase().startsWith("ban-ip ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     this.configManager.banIP(string4);
                     this.print(string3, new StringBuilder("Banning ip ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("pardon-ip ")) {
+                } else if (string2.toLowerCase().startsWith("pardon-ip ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     this.configManager.pardonIP(string4);
                     this.print(string3, new StringBuilder("Pardoning ip ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("ban ")) {
+                } else if (string2.toLowerCase().startsWith("ban ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     this.configManager.banPlayer(string4);
@@ -320,15 +296,13 @@
                     }
                     entityPlayerMP12.playerNetServerHandler.kickPlayer("Banned by admin");
-                }
-                else if (string2.toLowerCase().startsWith("pardon ")) {
+                } else if (string2.toLowerCase().startsWith("pardon ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     this.configManager.pardonPlayer(string4);
                     this.print(string3, new StringBuilder("Pardoning ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("kick ")) {
+                } else if (string2.toLowerCase().startsWith("kick ")) {
                     final String string4 = string2.substring(string2.indexOf(" ")).trim();
                     EntityPlayerMP entityPlayerMP12 = null;
                     for (int i13 = 0; i13 < this.configManager.playerEntities.size(); ++i13) {
-                        final EntityPlayerMP entityPlayerMP13 = (EntityPlayerMP)this.configManager.playerEntities.get(i13);
+                        final EntityPlayerMP entityPlayerMP13 = (EntityPlayerMP) this.configManager.playerEntities.get(i13);
                         if (entityPlayerMP13.username.equalsIgnoreCase(string4)) {
                             entityPlayerMP12 = entityPlayerMP13;
@@ -338,10 +312,8 @@
                         entityPlayerMP12.playerNetServerHandler.kickPlayer("Kicked by admin");
                         this.print(string3, new StringBuilder("Kicking ").append(entityPlayerMP12.username).toString());
-                    }
-                    else {
+                    } else {
                         iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string4).append(". No kick.").toString());
                     }
-                }
-                else if (string2.toLowerCase().startsWith("tp ")) {
+                } else if (string2.toLowerCase().startsWith("tp ")) {
                     final String[] string5 = string2.split(" ");
                     if (string5.length == 3) {
@@ -350,18 +322,14 @@
                         if (entityPlayerMP12 == null) {
                             iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string5[1]).append(". No tp.").toString());
-                        }
-                        else if (entityPlayerMP14 == null) {
+                        } else if (entityPlayerMP14 == null) {
                             iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string5[2]).append(". No tp.").toString());
-                        }
-                        else {
+                        } else {
                             entityPlayerMP12.playerNetServerHandler.teleportTo(entityPlayerMP14.posX, entityPlayerMP14.posY, entityPlayerMP14.posZ, entityPlayerMP14.rotationYaw, entityPlayerMP14.rotationPitch);
                             this.print(string3, new StringBuilder("Teleporting ").append(string5[1]).append(" to ").append(string5[2]).append(".").toString());
                         }
-                    }
-                    else {
+                    } else {
                         iCommandListener3.addHelpCommandMessage("Syntax error, please provice a source and a target.");
                     }
-                }
-                else if (string2.toLowerCase().startsWith("give ")) {
+                } else if (string2.toLowerCase().startsWith("give ")) {
                     final String[] string5 = string2.split(" ");
                     if (string5.length != 3 && string5.length != 4) {
@@ -386,23 +354,18 @@
                                 }
                                 entityPlayerMP14.dropPlayerItem(new ItemStack(i14, i15));
-                            }
-                            else {
+                            } else {
                                 iCommandListener3.addHelpCommandMessage(new StringBuilder("There's no item with id ").append(i14).toString());
                             }
-                        }
-                        catch (NumberFormatException numberFormatException10) {
+                        } catch (NumberFormatException numberFormatException10) {
                             iCommandListener3.addHelpCommandMessage(new StringBuilder("There's no item with id ").append(string5[2]).toString());
                         }
-                    }
-                    else {
+                    } else {
                         iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string6).toString());
                     }
-                }
-                else if (string2.toLowerCase().startsWith("say ")) {
+                } else if (string2.toLowerCase().startsWith("say ")) {
                     string2 = string2.substring(string2.indexOf(" ")).trim();
                     MinecraftServer.logger.info(new StringBuilder("[").append(string3).append("] ").append(string2).toString());
                     this.configManager.sendPacketToAllPlayers(new Packet3Chat(new StringBuilder("§d[Server] ").append(string2).toString()));
-                }
-                else if (string2.toLowerCase().startsWith("tell ")) {
+                } else if (string2.toLowerCase().startsWith("tell ")) {
                     final String[] string5 = string2.split(" ");
                     if (string5.length < 3) {
@@ -419,10 +382,8 @@
                     }
                     iCommandListener3.addHelpCommandMessage("There's no player by that name online.");
-                }
-                else {
+                } else {
                     MinecraftServer.logger.info("Unknown console command. Type \"help\" for help.");
                 }
-            }
-            else {
+            } else {
                 iCommandListener3.addHelpCommandMessage("To run the server without a gui, start it like this:");
                 iCommandListener3.addHelpCommandMessage("   java -Xmx1024M -Xms1024M -jar minecraft_server.jar nogui");
@@ -448,5 +409,5 @@
         }
     }
-    
+
     private void print(final String string1, final String string2) {
         final String string3 = String.valueOf(string1) + ": " + string2;
@@ -454,18 +415,17 @@
         MinecraftServer.logger.info(string3);
     }
-    
+
     private int parseInt(final String string, final int defaultValue) {
         try {
             return Integer.parseInt(string);
-        }
-        catch (NumberFormatException numberFormatException4) {
+        } catch (NumberFormatException numberFormatException4) {
             return defaultValue;
         }
     }
-    
+
     public void addToOnlinePlayerList(final IUpdatePlayerListBox playerListBox) {
         this.playersOnline.add(playerListBox);
     }
-    
+
     public static void main(final String[] args) {
         try {
@@ -475,24 +435,474 @@
             }
             new ThreadServerApplication("Server thread", minecraftServer1).start();
+        } catch (Exception exception2) {
+            MinecraftServer.logger.log(Level.SEVERE, "Failed to start the minecraft server", (Throwable) exception2);
         }
-        catch (Exception exception2) {
-            MinecraftServer.logger.log(Level.SEVERE, "Failed to start the minecraft server", (Throwable)exception2);
-        }
     }
-    
+
     public File getFile(final String fileName) {
         return new File(fileName);
     }
-    
+
     public void addHelpCommandMessage(final String helpCommandMessage) {
         MinecraftServer.logger.info(helpCommandMessage);
     }
-    
+
     public String getUsername() {
         return "CONSOLE";
     }
-    
+
     public static boolean isServerRunning(final MinecraftServer mcServer) {
         return mcServer.serverRunning;
     }
+
+    /**
+     * Gets the name of this server implementation
+     *
+     * @return name of this server implementation
+     */
+    @Override
+    public String getName() {
+        return "LilyBukkit";
+    }
+
+    /**
+     * Gets the version string of this server implementation.
+     *
+     * @return version of this server implementation
+     */
+    @Override
+    public String getVersion() {
+        return "1.0.0";
+    }
+
+    /**
+     * Gets a list of all currently logged in players
+     *
+     * @return An array of Players that are currently online
+     */
+    @Override
+    public Player[] getOnlinePlayers() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Get the maximum amount of players which can login to this server
+     *
+     * @return The amount of players this server allows
+     */
+    @Override
+    public int getMaxPlayers() {
+        return this.propertyManagerObj.getIntProperty(LBServerProperties.MAX_PLAYERS, 20);
+    }
+
+    /**
+     * Get the game port that the server runs on
+     *
+     * @return The port number of this server
+     */
+    @Override
+    public int getPort() {
+        return this.propertyManagerObj.getIntProperty(LBServerProperties.SERVER_PORT, 25565);
+    }
+
+    /**
+     * Get the view distance from this server.
+     *
+     * @return The view distance from this server.
+     */
+    @Override
+    public int getViewDistance() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Get the IP that this server is bound to or empty string if not specified
+     *
+     * @return The IP string that this server is bound to, otherwise empty string
+     */
+    @Override
+    public String getIp() {
+        return this.propertyManagerObj.getStringProperty(LBServerProperties.SERVER_IP, "");
+    }
+
+    /**
+     * Get the name of this server
+     *
+     * @return The name of this server
+     */
+    @Override
+    public String getServerName() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Get an ID of this server. The ID is a simple generally alphanumeric
+     * ID that can be used for uniquely identifying this server.
+     *
+     * @return The ID of this server
+     */
+    @Override
+    public String getServerId() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets whether this server allows the Nether or not.
+     *
+     * @return Whether this server allows the Nether or not.
+     */
+    @Override
+    public boolean getAllowNether() {
+        //TODO: Create Nether stub
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets whether this server has a whitelist or not.
+     *
+     * @return Whether this server has a whitelist or not.
+     */
+    @Override
+    public boolean hasWhitelist() {
+        return this.propertyManagerObj.getBooleanProperty(LBServerProperties.WHITELIST, false);
+    }
+
+    /**
+     * Broadcast a message to all players.
+     *
+     * @param message the message
+     * @return the number of players
+     */
+    @Override
+    public int broadcastMessage(String message) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the name of the update folder. The update folder is used to safely update
+     * plugins at the right moment on a plugin load.
+     *
+     * @return The name of the update folder
+     */
+    @Override
+    public String getUpdateFolder() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets a player object by the given username
+     * <p>
+     * This method may not return objects for offline players
+     *
+     * @param name Name to look up
+     * @return Player if it was found, otherwise null
+     */
+    @Override
+    public Player getPlayer(String name) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Attempts to match any players with the given name, and returns a list
+     * of all possibly matches
+     * <p>
+     * This list is not sorted in any particular order. If an exact match is found,
+     * the returned list will only contain a single result.
+     *
+     * @param name Name to match
+     * @return List of all possible players
+     */
+    @Override
+    public List<Player> matchPlayer(String name) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the PluginManager for interfacing with plugins
+     *
+     * @return PluginManager for this Server instance
+     */
+    @Override
+    public PluginManager getPluginManager() {
+        return new SimplePluginManager(this, LBPluginManager.getCommandMap());
+    }
+
+    /**
+     * Gets the Scheduler for managing scheduled events
+     *
+     * @return Scheduler for this Server instance
+     */
+    @Override
+    public BukkitScheduler getScheduler() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets a services manager
+     *
+     * @return Services manager
+     */
+    @Override
+    public ServicesManager getServicesManager() {
+        return new SimpleServicesManager();
+    }
+
+    /**
+     * Gets a list of all worlds on this server
+     *
+     * @return A list of worlds
+     */
+    @Override
+    public List<World> getWorlds() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Creates or loads a world with the given name.
+     * If the world is already loaded, it will just return the equivalent of
+     * getWorld(name)
+     *
+     * @param name        Name of the world to load
+     * @param environment Environment type of the world
+     * @return Newly created or loaded World
+     */
+    @Override
+    public World createWorld(String name, World.Environment environment) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Creates or loads a world with the given name.
+     * If the world is already loaded, it will just return the equivalent of
+     * getWorld(name)
+     *
+     * @param name        Name of the world to load
+     * @param environment Environment type of the world
+     * @param seed        Seed value to create the world with
+     * @return Newly created or loaded World
+     */
+    @Override
+    public World createWorld(String name, World.Environment environment, long seed) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Creates or loads a world with the given name.
+     * If the world is already loaded, it will just return the equivalent of
+     * getWorld(name)
+     *
+     * @param name        Name of the world to load
+     * @param environment Environment type of the world
+     * @param generator   ChunkGenerator to use in the construction of the new world
+     * @return Newly created or loaded World
+     */
+    @Override
+    public World createWorld(String name, World.Environment environment, ChunkGenerator generator) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Creates or loads a world with the given name.
+     * If the world is already loaded, it will just return the equivalent of
+     * getWorld(name)
+     *
+     * @param name        Name of the world to load
+     * @param environment Environment type of the world
+     * @param seed        Seed value to create the world with
+     * @param generator   ChunkGenerator to use in the construction of the new world
+     * @return Newly created or loaded World
+     */
+    @Override
+    public World createWorld(String name, World.Environment environment, long seed, ChunkGenerator generator) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Unloads a world with the given name.
+     *
+     * @param name Name of the world to unload
+     * @param save Whether to save the chunks before unloading.
+     * @return Whether the action was Successful
+     */
+    @Override
+    public boolean unloadWorld(String name, boolean save) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Unloads the given world.
+     *
+     * @param world The world to unload
+     * @param save  Whether to save the chunks before unloading.
+     * @return Whether the action was Successful
+     */
+    @Override
+    public boolean unloadWorld(World world, boolean save) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the world with the given name
+     *
+     * @param name Name of the world to retrieve
+     * @return World with the given name, or null if none exists
+     */
+    @Override
+    public World getWorld(String name) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the world from the given Unique ID
+     *
+     * @param uid Unique ID of the world to retrieve.
+     * @return World with the given Unique ID, or null if none exists.
+     */
+    @Override
+    public World getWorld(UUID uid) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the map from the given item ID.
+     *
+     * @param id ID of the map to get.
+     * @return The MapView if it exists, or null otherwise.
+     */
+    @Override
+    public MapView getMap(short id) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Create a new map with an automatically assigned ID.
+     *
+     * @param world The world the map will belong to.
+     * @return The MapView just created.
+     */
+    @Override
+    public MapView createMap(World world) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Reloads the server, refreshing settings and plugin information
+     */
+    @Override
+    public void reload() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Returns the primary logger associated with this server instance
+     *
+     * @return Logger associated with this server
+     */
+    @Override
+    public Logger getLogger() {
+        return logger;
+    }
+
+    /**
+     * Gets a {@link PluginCommand} with the given name or alias
+     *
+     * @param name Name of the command to retrieve
+     * @return PluginCommand if found, otherwise null
+     */
+    @Override
+    public PluginCommand getPluginCommand(String name) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Writes loaded players to disk
+     */
+    @Override
+    public void savePlayers() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Dispatches a command on the server, and executes it if found.
+     *
+     * @param sender
+     * @param commandLine
+     * @return targetFound returns false if no target is found.
+     * @throws CommandException Thrown when the executor for the given command fails with an unhandled exception
+     */
+    @Override
+    public boolean dispatchCommand(CommandSender sender, String commandLine) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Populates a given {@link ServerConfig} with values attributes to this server
+     *
+     * @param config ServerConfig to populate
+     */
+    @Override
+    public void configureDbConfig(ServerConfig config) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Adds a recipe to the crafting manager.
+     *
+     * @param recipe The recipe to add.
+     * @return True to indicate that the recipe was added.
+     */
+    @Override
+    public boolean addRecipe(Recipe recipe) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets a list of command aliases defined in the server properties.
+     *
+     * @return Map of aliases to command names
+     */
+    @Override
+    public Map<String, String[]> getCommandAliases() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets the radius, in blocks, around each worlds spawn point to protect
+     *
+     * @return Spawn radius, or 0 if none
+     */
+    @Override
+    public int getSpawnRadius() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Sets the radius, in blocks, around each worlds spawn point to protect
+     *
+     * @param value New spawn radius, or 0 if none
+     */
+    @Override
+    public void setSpawnRadius(int value) {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    /**
+     * Gets whether the Server is in online mode or not.
+     *
+     * @return Whether the server is in online mode.
+     */
+    @Override
+    public boolean getOnlineMode() {
+        return this.propertyManagerObj.getBooleanProperty(LBServerProperties.ONLINE_MODE, true);
+    }
+
+    /**
+     * Gets whether this server allows flying or not.
+     *
+     * @return Whether this server allows flying or not.
+     */
+    @Override
+    public boolean getAllowFlight() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
 }
