--- MinecraftServer.java
+++ MinecraftServer.java
@@ -1,44 +1,50 @@
 package net.minecraft.server;
 
-import net.minecraft.src.ThreadServerApplication;
-import net.minecraft.src.ServerGUI;
-import java.awt.GraphicsEnvironment;
-import net.minecraft.src.Packet3Chat;
-import net.minecraft.src.ItemStack;
-import net.minecraft.src.Item;
-import net.minecraft.src.EntityPlayerMP;
-import net.minecraft.src.ServerCommand;
-import java.util.Iterator;
+import jline.ConsoleReader;
+import joptsimple.OptionSet;
+import net.minecraft.src.AxisAlignedBB;
+import net.minecraft.src.ConsoleLogManager;
+import net.minecraft.src.EntityTracker;
+import net.minecraft.src.ICommandListener;
 import net.minecraft.src.IUpdatePlayerListBox;
-import net.minecraft.src.Packet;
+import net.minecraft.src.NetworkListenThread;
 import net.minecraft.src.Packet4UpdateTime;
+import net.minecraft.src.PropertyManager;
+import net.minecraft.src.ServerCommand;
+import net.minecraft.src.ServerConfigurationManager;
+import net.minecraft.src.ServerGUI;
+import net.minecraft.src.ThreadCommandReader;
+import net.minecraft.src.ThreadServerApplication;
+import net.minecraft.src.ThreadSleepForeverServer;
 import net.minecraft.src.Vec3D;
-import net.minecraft.src.AxisAlignedBB;
-import net.minecraft.src.IProgressUpdate;
-import net.minecraft.src.IWorldAccess;
 import net.minecraft.src.WorldManager;
+import net.minecraft.src.WorldServer;
+import org.bukkit.Bukkit;
+import org.bukkit.command.ConsoleCommandSender;
+import org.bukkit.craftbukkit.LoggerOutputStream;
+import org.bukkit.craftbukkit.util.ServerShutdownThread;
+import org.bukkit.event.server.ServerCommandEvent;
+import org.bukkit.event.world.WorldInitEvent;
+import org.bukkit.event.world.WorldLoadEvent;
+import org.bukkit.event.world.WorldSaveEvent;
+import org.bukkit.plugin.PluginLoadOrder;
+import ru.vladthemountain.lilybukkit.LilyBukkit;
+
+import java.awt.GraphicsEnvironment;
+import java.io.File;
 import java.io.IOException;
-import java.util.logging.Level;
+import java.io.PrintStream;
 import java.net.InetAddress;
-import java.io.File;
-import net.minecraft.src.ConsoleLogManager;
-import net.minecraft.src.ThreadCommandReader;
-import net.minecraft.src.ThreadSleepForeverServer;
-import java.util.Collections;
 import java.util.ArrayList;
-import java.util.Random;
-import net.minecraft.src.EntityTracker;
-import java.util.List;
-import net.minecraft.src.ServerConfigurationManager;
-import net.minecraft.src.WorldServer;
-import net.minecraft.src.PropertyManager;
-import net.minecraft.src.NetworkListenThread;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
+import java.util.Random;
+import java.util.logging.Level;
 import java.util.logging.Logger;
-import net.minecraft.src.ICommandListener;
 
 public class MinecraftServer implements ICommandListener, Runnable {
     public static Logger logger;
-    public static HashMap playerList;
+    public static HashMap<String, Integer> playerList;
     public NetworkListenThread networkServer;
     public PropertyManager propertyManagerObj;
@@ -50,6 +56,6 @@
     public String currentTask;
     public int percentDone;
-    private List playersOnline;
-    private List commands;
+    private List<IUpdatePlayerListBox> playersOnline;
+    private List<ServerCommand> commands;
     public EntityTracker entityTracker;
     public boolean onlineMode;
@@ -57,26 +63,62 @@
     public String serverId;
     private static Random rand;
-    
+
+    public static int[] getVersion() {
+        return new int[]{1, 1, 0, 5};
+    }
+
+    public static String getVersionTag() {
+        return "beta";
+    }
+
+    public static String getVersionString() {
+        int[] version = getVersion();
+        StringBuilder str = new StringBuilder();
+        for (int i = 0; i < version.length; i++) {
+            if (i != 0) str.append(".");
+            str.append(version[i]);
+        }
+        String tag = getVersionTag();
+        if (tag.length() > 0) str.append("-").append(tag);
+        return str.toString();
+    }
+
     static {
         MinecraftServer.logger = Logger.getLogger("Minecraft");
-        MinecraftServer.playerList = new HashMap();
+        MinecraftServer.playerList = new HashMap<>();
         MinecraftServer.rand = new Random(System.currentTimeMillis());
     }
-    
-    public MinecraftServer() {
+
+    public MinecraftServer(OptionSet opts) {
         this.serverRunning = true;
         this.serverStopped = false;
         this.deathTime = 0;
-        this.playersOnline = (List)new ArrayList();
-        this.commands = Collections.synchronizedList((List)new ArrayList());
+        this.playersOnline = new ArrayList<>();
+        this.commands = Collections.synchronizedList(new ArrayList<>());
+        // CraftBukkit start
+        this.options = opts;
+        // CraftBukkit end
         new ThreadSleepForeverServer(this);
+        // LilyBukkit start
+        this.lilyBukkit = new LilyBukkit(this);
+        this.consoleCommandSender = this.lilyBukkit.getConsoleSender();
+        try {
+            this.reader = new ConsoleReader();
+        } catch (IOException ex) {
+            Logger.getLogger(MinecraftServer.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        Runtime.getRuntime().addShutdownHook(new ServerShutdownThread(this));
     }
-    
+
     private boolean startServer() throws IOException {
         final ThreadCommandReader threadCommandReader1 = new ThreadCommandReader(this);
         threadCommandReader1.setDaemon(true);
         threadCommandReader1.start();
-        ConsoleLogManager.init();
-        MinecraftServer.logger.info("Starting minecraft server version 0.2.1");
+        ConsoleLogManager.init(this);
+        // CraftBukkit start
+        System.setOut(new PrintStream(new LoggerOutputStream(logger, Level.INFO), true));
+        System.setErr(new PrintStream(new LoggerOutputStream(logger, Level.SEVERE), true));
+        // CraftBukkit end
+        MinecraftServer.logger.info("[Minecraft] Starting Rosepad server version " + getVersionString() + ", running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " API version " + Bukkit.getBukkitVersion());
         if (Runtime.getRuntime().maxMemory() / 1024L / 1024L < 512L) {
             MinecraftServer.logger.warning("**** NOT ENOUGH RAM!");
@@ -84,5 +126,5 @@
         }
         MinecraftServer.logger.info("Loading properties");
-        this.propertyManagerObj = new PropertyManager(new File("server.properties"));
+        this.propertyManagerObj = new PropertyManager(this.options); // CraftBukkit
         final String string2 = this.propertyManagerObj.getStringProperty("server-ip", "");
         this.onlineMode = false;
@@ -93,12 +135,11 @@
         }
         final int i4 = this.propertyManagerObj.getIntProperty("server-port", 25565);
-        MinecraftServer.logger.info(new StringBuilder("Starting Minecraft server on ").append((string2.length() == 0) ? "*" : string2).append(":").append(i4).toString());
+        MinecraftServer.logger.info("Starting Minecraft server on " + ((string2.length() == 0) ? "*" : string2) + ":" + i4);
         this.serverId = Long.toHexString(MinecraftServer.rand.nextLong());
         try {
             this.networkServer = new NetworkListenThread(this, inetAddress3, i4);
-        }
-        catch (IOException iOException6) {
+        } catch (IOException iOException6) {
             MinecraftServer.logger.warning("**** FAILED TO BIND TO PORT!");
-            MinecraftServer.logger.log(Level.WARNING, new StringBuilder("The exception was: ").append(iOException6.toString()).toString());
+            MinecraftServer.logger.log(Level.WARNING, "The exception was: " + iOException6);
             MinecraftServer.logger.warning("Perhaps a server is already running on that port?");
             return false;
@@ -108,6 +149,5 @@
             MinecraftServer.logger.info("The server will authenticate usernames.");
             MinecraftServer.logger.info("Unauthenticated players may still join by default, and their abilities can be configured in the unauth.config file.");
-        }
-        else {
+        } else {
             MinecraftServer.logger.warning("**** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!");
             MinecraftServer.logger.warning("The server will make no attempt to authenticate usernames. Beware.");
@@ -118,10 +158,10 @@
         this.entityTracker = new EntityTracker(this);
         final String string3 = this.propertyManagerObj.getStringProperty("level-name", "world");
-        MinecraftServer.logger.info(new StringBuilder("Preparing level \"").append(string3).append("\"").toString());
+        MinecraftServer.logger.info("Preparing level \"" + string3 + "\"");
         this.initWorld(string3);
         MinecraftServer.logger.info("Done! For help, type \"help\" or \"?\"");
         return true;
     }
-    
+
     private void initWorld(final String worldName) {
         MinecraftServer.logger.info("Preparing start region");
@@ -140,26 +180,51 @@
         }
         this.clearCurrentTask();
+        // LilyBukkit start
+        if (!this.lilyBukkit.addExistingWorld(this.worldMngr)) {
+            MinecraftServer.logger.severe("[LilyBukkit] Couldn't get the current world, terminating to prevent bugs");
+            this.stop();
+        }
+        // Have to put them next to each other as the target LBWorld is created only in the expression above,
+        // otherwise events would be either called for a non-existent world or there would be several worlds existing
+        this.lilyBukkit.getPluginManager().callEvent(new WorldInitEvent(Bukkit.getWorld(worldName)));
+        this.lilyBukkit.getPluginManager().callEvent(new WorldLoadEvent(Bukkit.getWorld(worldName)));
+        // LilyBukkit stop
     }
-    
+
     private void outputPercentRemaining(final String currentTask, final int percent) {
         this.currentTask = currentTask;
         this.percentDone = percent;
-        System.out.println(String.valueOf(currentTask) + ": " + percent + "%");
+        System.out.print(currentTask + ": " + percent + "%");
     }
-    
+
     private void clearCurrentTask() {
         this.currentTask = null;
         this.percentDone = 0;
+
+        this.lilyBukkit.enablePluginsInOrder(PluginLoadOrder.POSTWORLD); // LilyBukkit
     }
-    
+
     private void save() {
         MinecraftServer.logger.info("Saving chunks");
+        WorldSaveEvent event = new WorldSaveEvent(lilyBukkit.getWorld(worldMngr.levelName));
+        this.lilyBukkit.getPluginManager().callEvent(event);
         this.worldMngr.saveWorld(true, null);
     }
-    
+
     private void stop() {
         MinecraftServer.logger.info("Stopping server");
+
+        // CraftBukkit start
+        if (this.lilyBukkit != null) {
+            this.lilyBukkit.getPluginManager().disablePlugins();
+        }
+        // CraftBukkit end
+
         if (this.configManager != null) {
             this.configManager.savePlayerStates();
+            // LilyBukkit - added
+            while (!this.configManager.playerEntities.isEmpty()) {
+                this.configManager.playerEntities.get(0).playerNetServerHandler.kickPlayer("Server closed");
+            }
         }
         if (this.worldMngr != null) {
@@ -167,9 +232,9 @@
         }
     }
-    
+
     public void stopRunning() {
         this.serverRunning = false;
     }
-    
+
     public void run() {
         try {
@@ -196,32 +261,27 @@
                     Thread.sleep(1L);
                 }
-            }
-            else {
+            } else {
                 while (this.serverRunning) {
                     this.commandLineParser();
                     try {
                         Thread.sleep(10L);
-                    }
-                    catch (InterruptedException interruptedException15) {
+                    } catch (InterruptedException interruptedException15) {
                         interruptedException15.printStackTrace();
                     }
                 }
             }
-        }
-        catch (Exception exception16) {
+        } catch (Exception exception16) {
             exception16.printStackTrace();
-            MinecraftServer.logger.log(Level.SEVERE, "Unexpected exception", (Throwable)exception16);
+            MinecraftServer.logger.log(Level.SEVERE, "[Minecraft] Unexpected exception", exception16);
             while (this.serverRunning) {
                 this.commandLineParser();
                 try {
                     Thread.sleep(10L);
-                }
-                catch (InterruptedException interruptedException16) {
+                } catch (InterruptedException interruptedException16) {
                     interruptedException16.printStackTrace();
                 }
             }
             return;
-        }
-        finally {
+        } finally {
             this.stop();
             this.serverStopped = true;
@@ -232,18 +292,17 @@
         System.exit(0);
     }
-    
+
     private void doTick() throws IOException {
-        final ArrayList arrayList1 = new ArrayList();
+        final ArrayList<String> arrayList1 = new ArrayList<>();
         for (final String string3 : MinecraftServer.playerList.keySet()) {
-            final int i4 = (int)MinecraftServer.playerList.get(string3);
+            final int i4 = MinecraftServer.playerList.get(string3);
             if (i4 > 0) {
                 MinecraftServer.playerList.put(string3, (i4 - 1));
-            }
-            else {
+            } else {
                 arrayList1.add(string3);
             }
         }
-        for (int i5 = 0; i5 < arrayList1.size(); ++i5) {
-            MinecraftServer.playerList.remove(arrayList1.get(i5));
+        for (Object o : arrayList1) {
+            MinecraftServer.playerList.remove(o);
         }
         AxisAlignedBB.clearBoundingBoxPool();
@@ -254,257 +313,85 @@
         }
         this.worldMngr.tick();
-        while (this.worldMngr.updatingLighting()) {}
+        while (this.worldMngr.updatingLighting()) {
+        }
         this.worldMngr.updateEntities();
         this.networkServer.handleNetworkListenThread();
         this.configManager.onTick();
         this.entityTracker.updateTrackedEntities();
-        for (int i5 = 0; i5 < this.playersOnline.size(); ++i5) {
-            ((IUpdatePlayerListBox)this.playersOnline.get(i5)).addAllPlayers();
+        for (IUpdatePlayerListBox iUpdatePlayerListBox : this.playersOnline) {
+            iUpdatePlayerListBox.addAllPlayers();
         }
         try {
             this.commandLineParser();
+        } catch (Exception exception5) {
+            MinecraftServer.logger.log(Level.WARNING, "Unexpected exception while parsing console command", exception5);
         }
-        catch (Exception exception5) {
-            MinecraftServer.logger.log(Level.WARNING, "Unexpected exception while parsing console command", (Throwable)exception5);
-        }
     }
-    
+
     public void addCommand(final String command, final ICommandListener commandListener) {
         this.commands.add(new ServerCommand(command, commandListener));
     }
-    
+
     public void commandLineParser() {
         while (this.commands.size() > 0) {
-            final ServerCommand serverCommand1 = (ServerCommand)this.commands.remove(0);
-            String string2 = serverCommand1.command;
-            final ICommandListener iCommandListener3 = serverCommand1.commandListener;
-            final String string3 = iCommandListener3.getUsername();
-            if (!string2.toLowerCase().startsWith("help") && !string2.toLowerCase().startsWith("?")) {
-                if (string2.toLowerCase().startsWith("list")) {
-                    iCommandListener3.addHelpCommandMessage(new StringBuilder("Connected players: ").append(this.configManager.getPlayerList()).toString());
-                }
-                else if (string2.toLowerCase().startsWith("stop")) {
-                    this.print(string3, "Stopping the server..");
-                    this.serverRunning = false;
-                }
-                else if (string2.toLowerCase().startsWith("save-all")) {
-                    this.print(string3, "Forcing save..");
-                    this.worldMngr.saveWorld(true, null);
-                    this.print(string3, "Save complete.");
-                }
-                else if (string2.toLowerCase().startsWith("save-off")) {
-                    this.print(string3, "Disabling level saving..");
-                    this.worldMngr.levelSaving = true;
-                }
-                else if (string2.toLowerCase().startsWith("save-on")) {
-                    this.print(string3, "Enabling level saving..");
-                    this.worldMngr.levelSaving = false;
-                }
-                else if (string2.toLowerCase().startsWith("op ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    this.configManager.opPlayer(string4);
-                    this.print(string3, new StringBuilder("Opping ").append(string4).toString());
-                    this.configManager.sendChatMessageToPlayer(string4, "§eYou are now op!");
-                }
-                else if (string2.toLowerCase().startsWith("deop ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    this.configManager.deopPlayer(string4);
-                    this.configManager.sendChatMessageToPlayer(string4, "§eYou are no longer op!");
-                    this.print(string3, new StringBuilder("De-opping ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("ban-ip ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    this.configManager.banIP(string4);
-                    this.print(string3, new StringBuilder("Banning ip ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("pardon-ip ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    this.configManager.pardonIP(string4);
-                    this.print(string3, new StringBuilder("Pardoning ip ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("ban ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    this.configManager.banPlayer(string4);
-                    this.print(string3, new StringBuilder("Banning ").append(string4).toString());
-                    final EntityPlayerMP entityPlayerMP12 = this.configManager.getPlayerEntity(string4);
-                    if (entityPlayerMP12 == null) {
-                        continue;
-                    }
-                    entityPlayerMP12.playerNetServerHandler.kickPlayer("Banned by admin");
-                }
-                else if (string2.toLowerCase().startsWith("pardon ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    this.configManager.pardonPlayer(string4);
-                    this.print(string3, new StringBuilder("Pardoning ").append(string4).toString());
-                }
-                else if (string2.toLowerCase().startsWith("kick ")) {
-                    final String string4 = string2.substring(string2.indexOf(" ")).trim();
-                    EntityPlayerMP entityPlayerMP12 = null;
-                    for (int i13 = 0; i13 < this.configManager.playerEntities.size(); ++i13) {
-                        final EntityPlayerMP entityPlayerMP13 = (EntityPlayerMP)this.configManager.playerEntities.get(i13);
-                        if (entityPlayerMP13.username.equalsIgnoreCase(string4)) {
-                            entityPlayerMP12 = entityPlayerMP13;
-                        }
-                    }
-                    if (entityPlayerMP12 != null) {
-                        entityPlayerMP12.playerNetServerHandler.kickPlayer("Kicked by admin");
-                        this.print(string3, new StringBuilder("Kicking ").append(entityPlayerMP12.username).toString());
-                    }
-                    else {
-                        iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string4).append(". No kick.").toString());
-                    }
-                }
-                else if (string2.toLowerCase().startsWith("tp ")) {
-                    final String[] string5 = string2.split(" ");
-                    if (string5.length == 3) {
-                        final EntityPlayerMP entityPlayerMP12 = this.configManager.getPlayerEntity(string5[1]);
-                        final EntityPlayerMP entityPlayerMP14 = this.configManager.getPlayerEntity(string5[2]);
-                        if (entityPlayerMP12 == null) {
-                            iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string5[1]).append(". No tp.").toString());
-                        }
-                        else if (entityPlayerMP14 == null) {
-                            iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string5[2]).append(". No tp.").toString());
-                        }
-                        else {
-                            entityPlayerMP12.playerNetServerHandler.teleportTo(entityPlayerMP14.posX, entityPlayerMP14.posY, entityPlayerMP14.posZ, entityPlayerMP14.rotationYaw, entityPlayerMP14.rotationPitch);
-                            this.print(string3, new StringBuilder("Teleporting ").append(string5[1]).append(" to ").append(string5[2]).append(".").toString());
-                        }
-                    }
-                    else {
-                        iCommandListener3.addHelpCommandMessage("Syntax error, please provice a source and a target.");
-                    }
-                }
-                else if (string2.toLowerCase().startsWith("give ")) {
-                    final String[] string5 = string2.split(" ");
-                    if (string5.length != 3 && string5.length != 4) {
-                        return;
-                    }
-                    final String string6 = string5[1];
-                    final EntityPlayerMP entityPlayerMP14 = this.configManager.getPlayerEntity(string6);
-                    if (entityPlayerMP14 != null) {
-                        try {
-                            final int i14 = Integer.parseInt(string5[2]);
-                            if (Item.itemsList[i14] != null) {
-                                this.print(string3, new StringBuilder("Giving ").append(entityPlayerMP14.username).append(" some ").append(i14).toString());
-                                int i15 = 1;
-                                if (string5.length > 3) {
-                                    i15 = this.parseInt(string5[3], 1);
-                                }
-                                if (i15 < 1) {
-                                    i15 = 1;
-                                }
-                                if (i15 > 64) {
-                                    i15 = 64;
-                                }
-                                entityPlayerMP14.dropPlayerItem(new ItemStack(i14, i15));
-                            }
-                            else {
-                                iCommandListener3.addHelpCommandMessage(new StringBuilder("There's no item with id ").append(i14).toString());
-                            }
-                        }
-                        catch (NumberFormatException numberFormatException10) {
-                            iCommandListener3.addHelpCommandMessage(new StringBuilder("There's no item with id ").append(string5[2]).toString());
-                        }
-                    }
-                    else {
-                        iCommandListener3.addHelpCommandMessage(new StringBuilder("Can't find user ").append(string6).toString());
-                    }
-                }
-                else if (string2.toLowerCase().startsWith("say ")) {
-                    string2 = string2.substring(string2.indexOf(" ")).trim();
-                    MinecraftServer.logger.info(new StringBuilder("[").append(string3).append("] ").append(string2).toString());
-                    this.configManager.sendPacketToAllPlayers(new Packet3Chat(new StringBuilder("§d[Server] ").append(string2).toString()));
-                }
-                else if (string2.toLowerCase().startsWith("tell ")) {
-                    final String[] string5 = string2.split(" ");
-                    if (string5.length < 3) {
-                        continue;
-                    }
-                    string2 = string2.substring(string2.indexOf(" ")).trim();
-                    string2 = string2.substring(string2.indexOf(" ")).trim();
-                    MinecraftServer.logger.info(new StringBuilder("[").append(string3).append("->").append(string5[1]).append("] ").append(string2).toString());
-                    this.configManager.sendPacketToAllPlayers(new Packet3Chat(new StringBuilder("§d[Server] ").append(string2).toString()));
-                    string2 = new StringBuilder("§7").append(string3).append(" whispers ").append(string2).toString();
-                    MinecraftServer.logger.info(string2);
-                    if (this.configManager.sendPacketToPlayer(string5[1], new Packet3Chat(string2))) {
-                        continue;
-                    }
-                    iCommandListener3.addHelpCommandMessage("There's no player by that name online.");
-                }
-                else {
-                    MinecraftServer.logger.info("Unknown console command. Type \"help\" for help.");
-                }
-            }
-            else {
-                iCommandListener3.addHelpCommandMessage("To run the server without a gui, start it like this:");
-                iCommandListener3.addHelpCommandMessage("   java -Xmx1024M -Xms1024M -jar minecraft_server.jar nogui");
-                iCommandListener3.addHelpCommandMessage("Console commands:");
-                iCommandListener3.addHelpCommandMessage("   help  or  ?               shows this message");
-                iCommandListener3.addHelpCommandMessage("   kick <player>             removes a player from the server");
-                iCommandListener3.addHelpCommandMessage("   ban <player>              bans a player from the server");
-                iCommandListener3.addHelpCommandMessage("   pardon <player>           pardons a banned player so that they can connect again");
-                iCommandListener3.addHelpCommandMessage("   ban-ip <ip>               bans an IP address from the server");
-                iCommandListener3.addHelpCommandMessage("   pardon-ip <ip>            pardons a banned IP address so that they can connect again");
-                iCommandListener3.addHelpCommandMessage("   op <player>               turns a player into an op");
-                iCommandListener3.addHelpCommandMessage("   deop <player>             removes op status from a player");
-                iCommandListener3.addHelpCommandMessage("   tp <player1> <player2>    moves one player to the same location as another player");
-                iCommandListener3.addHelpCommandMessage("   give <player> <id> [num]  gives a player a resource");
-                iCommandListener3.addHelpCommandMessage("   tell <player> <message>   sends a private message to a player");
-                iCommandListener3.addHelpCommandMessage("   stop                      gracefully stops the server");
-                iCommandListener3.addHelpCommandMessage("   save-all                  forces a server-wide level save");
-                iCommandListener3.addHelpCommandMessage("   save-off                  disables terrain saving (useful for backup scripts)");
-                iCommandListener3.addHelpCommandMessage("   save-on                   re-enables terrain saving");
-                iCommandListener3.addHelpCommandMessage("   list                      lists all currently connected players");
-                iCommandListener3.addHelpCommandMessage("   say <message>             broadcasts a message to all players");
-            }
+            final ServerCommand serverCommand1 = this.commands.remove(0);
+
+            this.lilyBukkit.getPluginManager().callEvent(new ServerCommandEvent(this.consoleCommandSender, serverCommand1.command));
+
+            if (this.lilyBukkit.getPlayer(serverCommand1.commandListener.getUsername()) != null)
+                this.lilyBukkit.dispatchCommand(this.lilyBukkit.getPlayer(serverCommand1.commandListener.getUsername()), serverCommand1.command);
+            else this.lilyBukkit.dispatchCommand(this.consoleCommandSender, serverCommand1.command);
         }
     }
-    
+
     private void print(final String string1, final String string2) {
-        final String string3 = String.valueOf(string1) + ": " + string2;
-        this.configManager.sendChatMessageToAllOps(new StringBuilder("§7(").append(string3).append(")").toString());
+        final String string3 = string1 + ": " + string2;
+        this.configManager.sendChatMessageToAllOps("§7(" + string3 + ")");
         MinecraftServer.logger.info(string3);
     }
-    
+
     private int parseInt(final String string, final int defaultValue) {
         try {
             return Integer.parseInt(string);
-        }
-        catch (NumberFormatException numberFormatException4) {
+        } catch (NumberFormatException numberFormatException4) {
             return defaultValue;
         }
     }
-    
+
     public void addToOnlinePlayerList(final IUpdatePlayerListBox playerListBox) {
         this.playersOnline.add(playerListBox);
     }
-    
-    public static void main(final String[] args) {
+
+    public static void main(final OptionSet options/*String[] args*/) {
         try {
-            final MinecraftServer minecraftServer1 = new MinecraftServer();
-            if (!GraphicsEnvironment.isHeadless() && (args.length <= 0 || !args[0].equals("nogui"))) {
+            final MinecraftServer minecraftServer1 = new MinecraftServer(options);
+            if (!GraphicsEnvironment.isHeadless() && !options.has("nogui")) {
                 ServerGUI.initGui(minecraftServer1);
             }
             new ThreadServerApplication("Server thread", minecraftServer1).start();
+        } catch (Exception exception2) {
+            MinecraftServer.logger.log(Level.SEVERE, "Failed to start the minecraft server", exception2);
         }
-        catch (Exception exception2) {
-            MinecraftServer.logger.log(Level.SEVERE, "Failed to start the minecraft server", (Throwable)exception2);
-        }
     }
-    
+
     public File getFile(final String fileName) {
         return new File(fileName);
     }
-    
+
     public void addHelpCommandMessage(final String helpCommandMessage) {
         MinecraftServer.logger.info(helpCommandMessage);
     }
-    
+
     public String getUsername() {
         return "CONSOLE";
     }
-    
+
     public static boolean isServerRunning(final MinecraftServer mcServer) {
         return mcServer.serverRunning;
     }
+
+    public LilyBukkit lilyBukkit; // LilyBukkit - added
+    public final ConsoleCommandSender consoleCommandSender;
+    public ConsoleReader reader;
+    public OptionSet options; //CraftBukkit
 }
